diff --git a/src/changes/changes.xml b/src/changes/changes.xml
index bc720f7596aaa1d39fcfbd1d31559591f25b1546..8a9861e1378737e691ab4328b66643e6bce514eb 100644
--- a/src/changes/changes.xml
+++ b/src/changes/changes.xml
@@ -46,6 +46,7 @@ The <action> type attribute can be add,update,fix,remove.
   <body>
 
   <release version="3.5" date="tba" description="New features including Java 9 detection">
+    <action issue="LANG-1255" type="add" dev="britter" due-to="Kaiyuan Wang">Add DateUtils.toCalendar(Date, TimeZone)</action>
     <action issue="LANG-1261" type="fix" dev="pschumacher" >ArrayUtils.contains returns false for instances of subtypes</action>
     <action issue="LANG-1197" type="update" dev="pschumacher" >Prepare Java 9 detection</action>
     <action issue="LANG-1252" type="fix" dev="chtompki" due-to="Rob Tompkins">Rename NumberUtils.isNumber, isCreatable to better reflect createNumber. Also, accommodated for "+" symbol as prefix in isCreatable and isNumber.</action>
diff --git a/src/main/java/org/apache/commons/lang3/time/DateUtils.java b/src/main/java/org/apache/commons/lang3/time/DateUtils.java
index c242bd587fc83ef5239155ed5f5cde24ef944ac1..8349c258315f7b03c30938daf46256257c4e345c 100644
--- a/src/main/java/org/apache/commons/lang3/time/DateUtils.java
+++ b/src/main/java/org/apache/commons/lang3/time/DateUtils.java
@@ -666,6 +666,20 @@ public class DateUtils {
         return c;
     }
     
+    //-----------------------------------------------------------------------
+    /**
+     * Converts a {@code Date} of a given {@code TimeZone} into a {@code Calendar}
+     * @param date the date to convert to a Calendar
+     * @param tz the time zone of the @{code date}
+     * @return the created Calendar
+     * @throws NullPointerException if {@code date} or {@code tz} is null
+     */
+    public static Calendar toCalendar(final Date date, final TimeZone tz) {
+    	final Calendar c = Calendar.getInstance(tz);
+    	c.setTime(date);
+    	return c;
+    }
+    
     //-----------------------------------------------------------------------
     /**
      * <p>Rounds a date, leaving the field specified as the most
