diff --git a/tests/test_chain.py b/tests/test_chain.py
index 6b2eae305f1266f948ad5f2f3e6671189f6b6ca3..702eaaa3e18edae197ec563497500cfb595f8441 100644
--- a/tests/test_chain.py
+++ b/tests/test_chain.py
@@ -189,7 +189,7 @@ def test_args_and_chain(runner):
     assert result.output.splitlines() == ["cli=", "a=", "b=", "c="]
 
 
-def test_multicommand_arg_behavior(runner):
+def test_group_arg_behavior(runner):
     with pytest.raises(RuntimeError):
 
         @click.group(chain=True)
@@ -219,7 +219,7 @@ def test_multicommand_arg_behavior(runner):
 
 
 @pytest.mark.xfail
-def test_multicommand_chaining(runner):
+def test_group_chaining(runner):
     @click.group(chain=True)
     def cli():
         debug()
diff --git a/tests/test_command_decorators.py b/tests/test_command_decorators.py
index cfdab3c31f944e8532899bda170e325b2b708504..5bd268c0a60be62ed5b8d667d61ac35bc4c2a6b8 100644
--- a/tests/test_command_decorators.py
+++ b/tests/test_command_decorators.py
@@ -1,3 +1,5 @@
+import pytest
+
 import click
 
 
@@ -69,3 +71,22 @@ def test_params_argument(runner):
     assert cli.params[1].name == "b"
     result = runner.invoke(cli, ["1", "2"])
     assert result.output == "1 2\n"
+
+
+@pytest.mark.parametrize(
+    "name",
+    [
+        "init_data",
+        "init_data_command",
+        "init_data_cmd",
+        "init_data_group",
+        "init_data_grp",
+    ],
+)
+def test_generate_name(name: str) -> None:
+    def f():
+        pass
+
+    f.__name__ = name
+    f = click.command(f)
+    assert f.name == "init-data"
diff --git a/tests/test_commands.py b/tests/test_commands.py
index ed9d96f3c5178d1f43bcadd8db89d3f3a3fd3f8b..5a56799adfd4b2c64998476a948c38910782a646 100644
--- a/tests/test_commands.py
+++ b/tests/test_commands.py
@@ -145,14 +145,14 @@ def test_group_with_args(runner, args, exit_code, expect):
     assert expect in result.output
 
 
-def test_base_command(runner):
+def test_custom_parser(runner):
     import optparse
 
     @click.group()
     def cli():
         pass
 
-    class OptParseCommand(click.BaseCommand):
+    class OptParseCommand(click.Command):
         def __init__(self, name, parser, callback):
             super().__init__(name)
             self.parser = parser
@@ -249,7 +249,7 @@ def test_other_command_invoke_with_defaults(runner):
     result = runner.invoke(cli, standalone_mode=False)
     # invoke should type cast default values, str becomes int, empty
     # multiple should be empty tuple instead of None
-    assert result.return_value == ("other-cmd", 42, 15, ())
+    assert result.return_value == ("other", 42, 15, ())
 
 
 def test_invoked_subcommand(runner):
diff --git a/tests/test_defaults.py b/tests/test_defaults.py
index 5ab05edb8002b2a93a49f3c8493c1ae8a4ba07ff..c93f9e68b8fb987952053c82011aeab1177f8514 100644
--- a/tests/test_defaults.py
+++ b/tests/test_defaults.py
@@ -5,7 +5,7 @@ def test_basic_defaults(runner):
     @click.command()
     @click.option("--foo", default=42, type=click.FLOAT)
     def cli(foo):
-        assert type(foo) is float  # noqa E721
+        assert isinstance(foo, float)
         click.echo(f"FOO:[{foo}]")
 
     result = runner.invoke(cli, [])
@@ -18,7 +18,7 @@ def test_multiple_defaults(runner):
     @click.option("--foo", default=[23, 42], type=click.FLOAT, multiple=True)
     def cli(foo):
         for item in foo:
-            assert type(item) is float  # noqa E721
+            assert isinstance(item, float)
             click.echo(item)
 
     result = runner.invoke(cli, [])
diff --git a/tests/test_imports.py b/tests/test_imports.py
index 453ffeee088935b6ccdaa6049c104bd753eba536..39dfbdbc24ac7d2b8c5c1f7b27e76b7acb52e19e 100644
--- a/tests/test_imports.py
+++ b/tests/test_imports.py
@@ -26,10 +26,12 @@ click.echo(json.dumps(rv))
 """
 
 ALLOWED_IMPORTS = {
+    "__future__",
     "weakref",
     "os",
     "struct",
     "collections",
+    "collections.abc",
     "sys",
     "contextlib",
     "functools",
diff --git a/tests/test_parser.py b/tests/test_parser.py
index f694916964828080bedc39024163451b2e9d292d..f2a3ad5931761ad0da0b4d3d8b7481bfd5f89ebb 100644
--- a/tests/test_parser.py
+++ b/tests/test_parser.py
@@ -1,8 +1,8 @@
 import pytest
 
 import click
-from click.parser import OptionParser
-from click.parser import split_arg_string
+from click.parser import _OptionParser
+from click.shell_completion import split_arg_string
 
 
 @pytest.mark.parametrize(
@@ -20,13 +20,13 @@ def test_split_arg_string(value, expect):
 
 
 def test_parser_default_prefixes():
-    parser = OptionParser()
+    parser = _OptionParser()
     assert parser._opt_prefixes == {"-", "--"}
 
 
 def test_parser_collects_prefixes():
     ctx = click.Context(click.Command("test"))
-    parser = OptionParser(ctx)
+    parser = _OptionParser(ctx)
     click.Option("+p", is_flag=True).add_to_parser(parser, ctx)
     click.Option("!e", is_flag=True).add_to_parser(parser, ctx)
     assert parser._opt_prefixes == {"-", "--", "+", "!"}
diff --git a/tests/test_testing.py b/tests/test_testing.py
index 4733221be4b2a54f1ed6508486aaad9476214d1f..2efef67dfc1323b5b677f1b4a62a03976ff2563a 100644
--- a/tests/test_testing.py
+++ b/tests/test_testing.py
@@ -321,7 +321,7 @@ def test_stderr():
     assert result_mix.stdout == "stdout\nstderr\n"
 
     with pytest.raises(ValueError):
-        result_mix.stderr  # noqa B018
+        assert result_mix.stderr  # noqa B018
 
     @click.command()
     def cli_empty_stderr():
diff --git a/tests/test_utils.py b/tests/test_utils.py
index fd099e9b20ca1cec3ff24e346d913525d44e76fe..b8b65e2956675bfffd9f46424607938a7e351623 100644
--- a/tests/test_utils.py
+++ b/tests/test_utils.py
@@ -241,7 +241,7 @@ def test_prompt_cast_default(capfd, monkeypatch):
     monkeypatch.setattr(sys, "stdin", StringIO("\n"))
     value = click.prompt("value", default="100", type=int)
     capfd.readouterr()
-    assert type(value) is int  # noqa E721
+    assert isinstance(value, int)
 
 
 @pytest.mark.skipif(WIN, reason="Test too complex to make work windows.")
