diff --git a/src/test/java/org/apache/commons/lang3/ArrayUtilsTest.java b/src/test/java/org/apache/commons/lang3/ArrayUtilsTest.java
index 609b169579c1d2ba4c4a7c2d3031699947eb3cf1..be3e119602714f53dc43221ea62e4396e7500260 100644
--- a/src/test/java/org/apache/commons/lang3/ArrayUtilsTest.java
+++ b/src/test/java/org/apache/commons/lang3/ArrayUtilsTest.java
@@ -2118,7 +2118,748 @@ public class ArrayUtilsTest  {
         assertEquals(null, array);
     }
     
+    //-----------------------------------------------------------------------
+    @Test
+    public void testSwapChar() {
+        char[] array = new char[] {1, 2, 3};
+        ArrayUtils.swap(array, 0, 2);
+        assertArrayEquals(new char[] {3, 2, 1}, array);
+        
+        array = new char[] {1, 2, 3};
+        ArrayUtils.swap(array, 0, 0);
+        assertArrayEquals(new char[] {1, 2, 3}, array);
+        
+        array = new char[] {1, 2, 3};
+        ArrayUtils.swap(array, 1, 0);
+        assertArrayEquals(new char[] {2, 1, 3}, array);
+    }
+    
+    @Test
+    public void testSwapCharRange() {
+        char[] array = new char[] {1, 2, 3, 4};
+        ArrayUtils.swap(array, 0, 2, 2);
+        assertEquals(3, array[0]);
+        assertEquals(4, array[1]);
+        assertEquals(1, array[2]);
+        assertEquals(2, array[3]);
+    }
+
+    @Test(expected = ArrayIndexOutOfBoundsException.class)
+    public void testSwapCharOutOfRange() {
+        char[] array = new char[] {1, 2, 3};
+        ArrayUtils.swap(array, 0, 3);
+    }
+
+    @Test(expected = ArrayIndexOutOfBoundsException.class)
+    public void testSwapCharOutOfRangeLen() {
+        char[] array = new char[] {1, 2, 3};
+        ArrayUtils.swap(array, 0, 2, 2);
+    }
+    
+    @Test
+    public void testSwapFloat() {
+        float[] array = new float[] {1, 2, 3};
+        ArrayUtils.swap(array, 0, 2);
+        assertEquals(3, array[0], 0);
+        assertEquals(2, array[1], 0);
+        assertEquals(1, array[2], 0);
+    }
+
+    @Test
+    public void testSwapFloatRange() {
+        float[] array = new float[] {1, 2, 3, 4};
+        ArrayUtils.swap(array, 0, 2, 2);
+        assertEquals(3, array[0], 0);
+        assertEquals(4, array[1], 0);
+        assertEquals(1, array[2], 0);
+        assertEquals(2, array[3], 0);
+    }
+
+    @Test(expected = ArrayIndexOutOfBoundsException.class)
+    public void testSwapFloatOutOfRange() {
+        float[] array = new float[] {1, 2, 3};
+        ArrayUtils.swap(array, 0, 3);
+    }
+
+    @Test(expected = ArrayIndexOutOfBoundsException.class)
+    public void testSwapFloatOutOfRangeLen() {
+        float[] array = new float[] {1, 2, 3};
+        ArrayUtils.swap(array, 0, 2, 2);
+    }
+    
+    @Test
+    public void testSwapDouble() {
+        double[] array = new double[] {1, 2, 3};
+        ArrayUtils.swap(array, 0, 2);
+        assertEquals(3, array[0], 0);
+        assertEquals(2, array[1], 0);
+        assertEquals(1, array[2], 0);
+    }
+
+    @Test
+    public void testSwapDoubleRange() {
+        double[] array = new double[] {1, 2, 3, 4};
+        ArrayUtils.swap(array, 0, 2, 2);
+        assertEquals(3, array[0], 0);
+        assertEquals(4, array[1], 0);
+        assertEquals(1, array[2], 0);
+        assertEquals(2, array[3], 0);
+    }
+
+    @Test(expected = ArrayIndexOutOfBoundsException.class)
+    public void testSwapDoubleOutOfRange() {
+        double[] array = new double[] {1, 2, 3};
+        ArrayUtils.swap(array, 0, 3);
+    }
+
+    @Test(expected = ArrayIndexOutOfBoundsException.class)
+    public void testSwapDoubleOutOfRangeLen() {
+        double[] array = new double[] {1, 2, 3};
+        ArrayUtils.swap(array, 0, 2, 2);
+    }
+    
+    @Test
+    public void testSwapInt() {
+        int[] array = new int[] {1, 2, 3};
+        ArrayUtils.swap(array, 0, 2);
+        assertEquals(3, array[0]);
+        assertEquals(2, array[1]);
+        assertEquals(1, array[2]);
+    }
+
+    @Test
+    public void testSwapIntRange() {
+        int[] array = new int[] {1, 2, 3, 4};
+        ArrayUtils.swap(array, 0, 2, 2);
+        assertEquals(3, array[0]);
+        assertEquals(4, array[1]);
+        assertEquals(1, array[2]);
+        assertEquals(2, array[3]);
+    }
+    
+    @Test
+    public void testSwapIntExchangedOffsets() {
+        int[] array;
+        array = new int[] {1, 2, 3};
+        ArrayUtils.swap(array, 0, 1, 2);
+        assertArrayEquals(new int[] {2, 3, 1}, array);
+        
+        array = new int[] {1, 2, 3};
+        ArrayUtils.swap(array, 1, 0, 2);
+        assertArrayEquals(new int[] {2, 3, 1}, array);
+    }
+
+    @Test(expected = ArrayIndexOutOfBoundsException.class)
+    public void testSwapIntOutOfRange() {
+        int[] array = new int[] {1, 2, 3};
+        ArrayUtils.swap(array, 0, 3);
+    }
+
+    @Test(expected = ArrayIndexOutOfBoundsException.class)
+    public void testSwapIntOutOfRangeLen() {
+        int[] array = new int[] {1, 2, 3};
+        ArrayUtils.swap(array, 0, 2, 2);
+    }
+    
+    @Test
+    public void testSwapLong() {
+        long[] array = new long[] {1, 2, 3};
+        ArrayUtils.swap(array, 0, 2);
+        assertEquals(3, array[0]);
+        assertEquals(2, array[1]);
+        assertEquals(1, array[2]);
+    }
+
+    @Test
+    public void testSwapLongRange() {
+        long[] array = new long[] {1, 2, 3, 4};
+        ArrayUtils.swap(array, 0, 2, 2);
+        assertEquals(3, array[0]);
+        assertEquals(4, array[1]);
+        assertEquals(1, array[2]);
+        assertEquals(2, array[3]);
+    }
+
+    @Test(expected = ArrayIndexOutOfBoundsException.class)
+    public void testSwapLongOutOfRange() {
+        long[] array = new long[] {1, 2, 3};
+        ArrayUtils.swap(array, 0, 3);
+    }
+
+    @Test(expected = ArrayIndexOutOfBoundsException.class)
+    public void testSwapLongOutOfRangeLen() {
+        long[] array = new long[] {1, 2, 3};
+        ArrayUtils.swap(array, 0, 2, 2);
+    }
+
+    @Test
+    public void testSwapObject() {
+        String[] array = new String[] {"1", "2", "3"};
+        ArrayUtils.swap(array, 0, 2);
+        assertEquals("3", array[0]);
+        assertEquals("2", array[1]);
+        assertEquals("1", array[2]);
+    }
+
+    @Test
+    public void testSwapObjectRange() {
+        String[] array = new String[] {"1", "2", "3", "4"};
+        ArrayUtils.swap(array, 0, 2, 2);
+        assertEquals("3", array[0]);
+        assertEquals("4", array[1]);
+        assertEquals("1", array[2]);
+        assertEquals("2", array[3]);
+    }
+
+    @Test(expected = ArrayIndexOutOfBoundsException.class)
+    public void testSwapObjectOutOfRange() {
+        String[] array = new String[] {"1", "2", "3"};
+        ArrayUtils.swap(array, 0, 3);
+    }
+
+    @Test(expected = ArrayIndexOutOfBoundsException.class)
+    public void testSwapObjectOutOfRangeLen() {
+        String[] array = new String[] {"1", "2", "3"};
+        ArrayUtils.swap(array, 0, 2, 2);
+    }
+
+    //-----------------------------------------------------------------------
+    @Test
+    public void testShiftDouble() {
+        double[] array = new double[] {1, 2, 3, 4};
+        ArrayUtils.shift(array, 1);
+        assertEquals(4, array[0], 0);
+        assertEquals(1, array[1], 0);
+        assertEquals(2, array[2], 0);
+        assertEquals(3, array[3], 0);
+        ArrayUtils.shift(array, -1);
+        assertEquals(1, array[0], 0);
+        assertEquals(2, array[1], 0);
+        assertEquals(3, array[2], 0);
+        assertEquals(4, array[3], 0);
+        ArrayUtils.shift(array, 5);
+        assertEquals(4, array[0], 0);
+        assertEquals(1, array[1], 0);
+        assertEquals(2, array[2], 0);
+        assertEquals(3, array[3], 0);
+        ArrayUtils.shift(array, -3);
+        assertEquals(3, array[0], 0);
+        assertEquals(4, array[1], 0);
+        assertEquals(1, array[2], 0);
+        assertEquals(2, array[3], 0);
+    }
+
+    @Test
+    public void testShiftRangeDouble() {
+        double[] array = new double[] {1, 2, 3, 4, 5};
+        ArrayUtils.shift(array, 1, 3, 1);
+        assertEquals(1, array[0], 0);
+        assertEquals(3, array[1], 0);
+        assertEquals(2, array[2], 0);
+        assertEquals(4, array[3], 0);
+        assertEquals(5, array[4], 0);
+        ArrayUtils.shift(array, 1, 4, 2);
+        assertEquals(1, array[0], 0);
+        assertEquals(2, array[1], 0);
+        assertEquals(4, array[2], 0);
+        assertEquals(3, array[3], 0);
+        assertEquals(5, array[4], 0);
+    }
+
+    @Test
+    public void testShiftRangeNoElemDouble() {
+        double[] array = new double[] {1, 2, 3, 4};
+        ArrayUtils.shift(array, 1, 1, 1);
+        assertEquals(1, array[0], 0);
+        assertEquals(2, array[1], 0);
+        assertEquals(3, array[2], 0);
+        assertEquals(4, array[3], 0);
+    }
+
+    @Test
+    public void testShiftAllDouble() {
+        double[] array = new double[] {1, 2, 3, 4};
+        ArrayUtils.shift(array, 4);
+        assertEquals(1, array[0], 0);
+        assertEquals(2, array[1], 0);
+        assertEquals(3, array[2], 0);
+        assertEquals(4, array[3], 0);
+        ArrayUtils.shift(array, -4);
+        assertEquals(1, array[0], 0);
+        assertEquals(2, array[1], 0);
+        assertEquals(3, array[2], 0);
+        assertEquals(4, array[3], 0);
+    }
+    
+    @Test
+    public void testShiftFloat() {
+        float[] array = new float[] {1, 2, 3, 4};
+        ArrayUtils.shift(array, 1);
+        assertEquals(4, array[0], 0);
+        assertEquals(1, array[1], 0);
+        assertEquals(2, array[2], 0);
+        assertEquals(3, array[3], 0);
+        ArrayUtils.shift(array, -1);
+        assertEquals(1, array[0], 0);
+        assertEquals(2, array[1], 0);
+        assertEquals(3, array[2], 0);
+        assertEquals(4, array[3], 0);
+        ArrayUtils.shift(array, 5);
+        assertEquals(4, array[0], 0);
+        assertEquals(1, array[1], 0);
+        assertEquals(2, array[2], 0);
+        assertEquals(3, array[3], 0);
+        ArrayUtils.shift(array, -3);
+        assertEquals(3, array[0], 0);
+        assertEquals(4, array[1], 0);
+        assertEquals(1, array[2], 0);
+        assertEquals(2, array[3], 0);
+    }
+
+    @Test
+    public void testShiftRangeFloat() {
+        float[] array = new float[] {1, 2, 3, 4, 5};
+        ArrayUtils.shift(array, 1, 3, 1);
+        assertEquals(1, array[0], 0);
+        assertEquals(3, array[1], 0);
+        assertEquals(2, array[2], 0);
+        assertEquals(4, array[3], 0);
+        assertEquals(5, array[4], 0);
+        ArrayUtils.shift(array, 1, 4, 2);
+        assertEquals(1, array[0], 0);
+        assertEquals(2, array[1], 0);
+        assertEquals(4, array[2], 0);
+        assertEquals(3, array[3], 0);
+        assertEquals(5, array[4], 0);
+    }
+
+    @Test
+    public void testShiftRangeNoElemFloat() {
+        float[] array = new float[] {1, 2, 3, 4};
+        ArrayUtils.shift(array, 1, 1, 1);
+        assertEquals(1, array[0], 0);
+        assertEquals(2, array[1], 0);
+        assertEquals(3, array[2], 0);
+        assertEquals(4, array[3], 0);
+    }
+
+    @Test
+    public void testShiftAllFloat() {
+        float[] array = new float[] {1, 2, 3, 4};
+        ArrayUtils.shift(array, 4);
+        assertEquals(1, array[0], 0);
+        assertEquals(2, array[1], 0);
+        assertEquals(3, array[2], 0);
+        assertEquals(4, array[3], 0);
+        ArrayUtils.shift(array, -4);
+        assertEquals(1, array[0], 0);
+        assertEquals(2, array[1], 0);
+        assertEquals(3, array[2], 0);
+        assertEquals(4, array[3], 0);
+    }
+    
+    @Test
+    public void testShiftShort() {
+        short[] array = new short[] {1, 2, 3, 4};
+        ArrayUtils.shift(array, 1);
+        assertEquals(4, array[0]);
+        assertEquals(1, array[1]);
+        assertEquals(2, array[2]);
+        assertEquals(3, array[3]);
+        ArrayUtils.shift(array, -1);
+        assertEquals(1, array[0]);
+        assertEquals(2, array[1]);
+        assertEquals(3, array[2]);
+        assertEquals(4, array[3]);
+        ArrayUtils.shift(array, 5);
+        assertEquals(4, array[0]);
+        assertEquals(1, array[1]);
+        assertEquals(2, array[2]);
+        assertEquals(3, array[3]);
+        ArrayUtils.shift(array, -3);
+        assertEquals(3, array[0]);
+        assertEquals(4, array[1]);
+        assertEquals(1, array[2]);
+        assertEquals(2, array[3]);
+    }
+
+    @Test
+    public void testShiftRangeShort() {
+        short[] array = new short[] {1, 2, 3, 4, 5};
+        ArrayUtils.shift(array, 1, 3, 1);
+        assertEquals(1, array[0]);
+        assertEquals(3, array[1]);
+        assertEquals(2, array[2]);
+        assertEquals(4, array[3]);
+        assertEquals(5, array[4]);
+        ArrayUtils.shift(array, 1, 4, 2);
+        assertEquals(1, array[0]);
+        assertEquals(2, array[1]);
+        assertEquals(4, array[2]);
+        assertEquals(3, array[3]);
+        assertEquals(5, array[4]);
+    }
+
+    @Test
+    public void testShiftRangeNoElemShort() {
+        short[] array = new short[] {1, 2, 3, 4};
+        ArrayUtils.shift(array, 1, 1, 1);
+        assertEquals(1, array[0]);
+        assertEquals(2, array[1]);
+        assertEquals(3, array[2]);
+        assertEquals(4, array[3]);
+    }
+
+    @Test
+    public void testShiftAllShort() {
+        short[] array = new short[] {1, 2, 3, 4};
+        ArrayUtils.shift(array, 4);
+        assertEquals(1, array[0]);
+        assertEquals(2, array[1]);
+        assertEquals(3, array[2]);
+        assertEquals(4, array[3]);
+        ArrayUtils.shift(array, -4);
+        assertEquals(1, array[0]);
+        assertEquals(2, array[1]);
+        assertEquals(3, array[2]);
+        assertEquals(4, array[3]);
+    }
+    
+    @Test
+    public void testShiftByte() {
+        byte[] array = new byte[] {1, 2, 3, 4};
+        ArrayUtils.shift(array, 1);
+        assertEquals(4, array[0]);
+        assertEquals(1, array[1]);
+        assertEquals(2, array[2]);
+        assertEquals(3, array[3]);
+        ArrayUtils.shift(array, -1);
+        assertEquals(1, array[0]);
+        assertEquals(2, array[1]);
+        assertEquals(3, array[2]);
+        assertEquals(4, array[3]);
+        ArrayUtils.shift(array, 5);
+        assertEquals(4, array[0]);
+        assertEquals(1, array[1]);
+        assertEquals(2, array[2]);
+        assertEquals(3, array[3]);
+        ArrayUtils.shift(array, -3);
+        assertEquals(3, array[0]);
+        assertEquals(4, array[1]);
+        assertEquals(1, array[2]);
+        assertEquals(2, array[3]);
+    }
+
+    @Test
+    public void testShiftRangeByte() {
+        byte[] array = new byte[] {1, 2, 3, 4, 5};
+        ArrayUtils.shift(array, 1, 3, 1);
+        assertEquals(1, array[0]);
+        assertEquals(3, array[1]);
+        assertEquals(2, array[2]);
+        assertEquals(4, array[3]);
+        assertEquals(5, array[4]);
+        ArrayUtils.shift(array, 1, 4, 2);
+        assertEquals(1, array[0]);
+        assertEquals(2, array[1]);
+        assertEquals(4, array[2]);
+        assertEquals(3, array[3]);
+        assertEquals(5, array[4]);
+    }
+
+    @Test
+    public void testShiftRangeNoElemByte() {
+        byte[] array = new byte[] {1, 2, 3, 4};
+        ArrayUtils.shift(array, 1, 1, 1);
+        assertEquals(1, array[0]);
+        assertEquals(2, array[1]);
+        assertEquals(3, array[2]);
+        assertEquals(4, array[3]);
+    }
+
+    @Test
+    public void testShiftAllByte() {
+        byte[] array = new byte[] {1, 2, 3, 4};
+        ArrayUtils.shift(array, 4);
+        assertEquals(1, array[0]);
+        assertEquals(2, array[1]);
+        assertEquals(3, array[2]);
+        assertEquals(4, array[3]);
+        ArrayUtils.shift(array, -4);
+        assertEquals(1, array[0]);
+        assertEquals(2, array[1]);
+        assertEquals(3, array[2]);
+        assertEquals(4, array[3]);
+    }
+    
+    @Test
+    public void testShiftChar() {
+        char[] array = new char[] {1, 2, 3, 4};
+        ArrayUtils.shift(array, 1);
+        assertEquals(4, array[0]);
+        assertEquals(1, array[1]);
+        assertEquals(2, array[2]);
+        assertEquals(3, array[3]);
+        ArrayUtils.shift(array, -1);
+        assertEquals(1, array[0]);
+        assertEquals(2, array[1]);
+        assertEquals(3, array[2]);
+        assertEquals(4, array[3]);
+        ArrayUtils.shift(array, 5);
+        assertEquals(4, array[0]);
+        assertEquals(1, array[1]);
+        assertEquals(2, array[2]);
+        assertEquals(3, array[3]);
+        ArrayUtils.shift(array, -3);
+        assertEquals(3, array[0]);
+        assertEquals(4, array[1]);
+        assertEquals(1, array[2]);
+        assertEquals(2, array[3]);
+    }
+
+    @Test
+    public void testShiftRangeChar() {
+        char[] array = new char[] {1, 2, 3, 4, 5};
+        ArrayUtils.shift(array, 1, 3, 1);
+        assertEquals(1, array[0]);
+        assertEquals(3, array[1]);
+        assertEquals(2, array[2]);
+        assertEquals(4, array[3]);
+        assertEquals(5, array[4]);
+        ArrayUtils.shift(array, 1, 4, 2);
+        assertEquals(1, array[0]);
+        assertEquals(2, array[1]);
+        assertEquals(4, array[2]);
+        assertEquals(3, array[3]);
+        assertEquals(5, array[4]);
+    }
+
+    @Test
+    public void testShiftRangeNoElemChar() {
+        char[] array = new char[] {1, 2, 3, 4};
+        ArrayUtils.shift(array, 1, 1, 1);
+        assertEquals(1, array[0]);
+        assertEquals(2, array[1]);
+        assertEquals(3, array[2]);
+        assertEquals(4, array[3]);
+    }
+
+    @Test
+    public void testShiftAllChar() {
+        char[] array = new char[] {1, 2, 3, 4};
+        ArrayUtils.shift(array, 4);
+        assertEquals(1, array[0]);
+        assertEquals(2, array[1]);
+        assertEquals(3, array[2]);
+        assertEquals(4, array[3]);
+        ArrayUtils.shift(array, -4);
+        assertEquals(1, array[0]);
+        assertEquals(2, array[1]);
+        assertEquals(3, array[2]);
+        assertEquals(4, array[3]);
+    }
+    
+    @Test
+    public void testShiftLong() {
+        long[] array = new long[] {1, 2, 3, 4};
+        ArrayUtils.shift(array, 1);
+        assertEquals(4, array[0]);
+        assertEquals(1, array[1]);
+        assertEquals(2, array[2]);
+        assertEquals(3, array[3]);
+        ArrayUtils.shift(array, -1);
+        assertEquals(1, array[0]);
+        assertEquals(2, array[1]);
+        assertEquals(3, array[2]);
+        assertEquals(4, array[3]);
+        ArrayUtils.shift(array, 5);
+        assertEquals(4, array[0]);
+        assertEquals(1, array[1]);
+        assertEquals(2, array[2]);
+        assertEquals(3, array[3]);
+        ArrayUtils.shift(array, -3);
+        assertEquals(3, array[0]);
+        assertEquals(4, array[1]);
+        assertEquals(1, array[2]);
+        assertEquals(2, array[3]);
+    }
+
+    @Test
+    public void testShiftRangeLong() {
+        long[] array = new long[] {1, 2, 3, 4, 5};
+        ArrayUtils.shift(array, 1, 3, 1);
+        assertEquals(1, array[0]);
+        assertEquals(3, array[1]);
+        assertEquals(2, array[2]);
+        assertEquals(4, array[3]);
+        assertEquals(5, array[4]);
+        ArrayUtils.shift(array, 1, 4, 2);
+        assertEquals(1, array[0]);
+        assertEquals(2, array[1]);
+        assertEquals(4, array[2]);
+        assertEquals(3, array[3]);
+        assertEquals(5, array[4]);
+    }
+
+    @Test
+    public void testShiftRangeNoElemLong() {
+        long[] array = new long[] {1, 2, 3, 4};
+        ArrayUtils.shift(array, 1, 1, 1);
+        assertEquals(1, array[0]);
+        assertEquals(2, array[1]);
+        assertEquals(3, array[2]);
+        assertEquals(4, array[3]);
+    }
+
+    @Test
+    public void testShiftAllLong() {
+        long[] array = new long[] {1, 2, 3, 4};
+        ArrayUtils.shift(array, 4);
+        assertEquals(1, array[0]);
+        assertEquals(2, array[1]);
+        assertEquals(3, array[2]);
+        assertEquals(4, array[3]);
+        ArrayUtils.shift(array, -4);
+        assertEquals(1, array[0]);
+        assertEquals(2, array[1]);
+        assertEquals(3, array[2]);
+        assertEquals(4, array[3]);
+    }
+    
+    @Test
+    public void testShiftInt() {
+        int[] array = new int[] {1, 2, 3, 4};
+        ArrayUtils.shift(array, 1);
+        assertEquals(4, array[0]);
+        assertEquals(1, array[1]);
+        assertEquals(2, array[2]);
+        assertEquals(3, array[3]);
+        ArrayUtils.shift(array, -1);
+        assertEquals(1, array[0]);
+        assertEquals(2, array[1]);
+        assertEquals(3, array[2]);
+        assertEquals(4, array[3]);
+        ArrayUtils.shift(array, 5);
+        assertEquals(4, array[0]);
+        assertEquals(1, array[1]);
+        assertEquals(2, array[2]);
+        assertEquals(3, array[3]);
+        ArrayUtils.shift(array, -3);
+        assertEquals(3, array[0]);
+        assertEquals(4, array[1]);
+        assertEquals(1, array[2]);
+        assertEquals(2, array[3]);
+    }
 
+    @Test
+    public void testShiftRangeInt() {
+        int[] array = new int[] {1, 2, 3, 4, 5};
+        ArrayUtils.shift(array, 1, 3, 1);
+        assertEquals(1, array[0]);
+        assertEquals(3, array[1]);
+        assertEquals(2, array[2]);
+        assertEquals(4, array[3]);
+        assertEquals(5, array[4]);
+        ArrayUtils.shift(array, 1, 4, 2);
+        assertEquals(1, array[0]);
+        assertEquals(2, array[1]);
+        assertEquals(4, array[2]);
+        assertEquals(3, array[3]);
+        assertEquals(5, array[4]);
+    }
+
+    @Test
+    public void testShiftRangeNoElemInt() {
+        int[] array = new int[] {1, 2, 3, 4};
+        ArrayUtils.shift(array, 1, 1, 1);
+        assertEquals(1, array[0]);
+        assertEquals(2, array[1]);
+        assertEquals(3, array[2]);
+        assertEquals(4, array[3]);
+    }
+
+    @Test
+    public void testShiftAllInt() {
+        int[] array = new int[] {1, 2, 3, 4};
+        ArrayUtils.shift(array, 4);
+        assertEquals(1, array[0]);
+        assertEquals(2, array[1]);
+        assertEquals(3, array[2]);
+        assertEquals(4, array[3]);
+        ArrayUtils.shift(array, -4);
+        assertEquals(1, array[0]);
+        assertEquals(2, array[1]);
+        assertEquals(3, array[2]);
+        assertEquals(4, array[3]);
+    }
+
+    @Test
+    public void testShiftObject() {
+        String[] array = new String[] {"1", "2", "3", "4"};
+        ArrayUtils.shift(array, 1);
+        assertEquals("4", array[0]);
+        assertEquals("1", array[1]);
+        assertEquals("2", array[2]);
+        assertEquals("3", array[3]);
+        ArrayUtils.shift(array, -1);
+        assertEquals("1", array[0]);
+        assertEquals("2", array[1]);
+        assertEquals("3", array[2]);
+        assertEquals("4", array[3]);
+        ArrayUtils.shift(array, 5);
+        assertEquals("4", array[0]);
+        assertEquals("1", array[1]);
+        assertEquals("2", array[2]);
+        assertEquals("3", array[3]);
+        ArrayUtils.shift(array, -3);
+        assertEquals("3", array[0]);
+        assertEquals("4", array[1]);
+        assertEquals("1", array[2]);
+        assertEquals("2", array[3]);
+    }
+
+    @Test
+    public void testShiftRangeObject() {
+        String[] array = new String[] {"1", "2", "3", "4", "5"};
+        ArrayUtils.shift(array, 1, 3, 1);
+        assertEquals("1", array[0]);
+        assertEquals("3", array[1]);
+        assertEquals("2", array[2]);
+        assertEquals("4", array[3]);
+        assertEquals("5", array[4]);
+        ArrayUtils.shift(array, 1, 4, 2);
+        assertEquals("1", array[0]);
+        assertEquals("2", array[1]);
+        assertEquals("4", array[2]);
+        assertEquals("3", array[3]);
+        assertEquals("5", array[4]);
+    }
+
+    @Test
+    public void testShiftRangeNoElemObject() {
+        String[] array = new String[] {"1", "2", "3", "4"};
+        ArrayUtils.shift(array, 1, 1, 1);
+        assertEquals("1", array[0]);
+        assertEquals("2", array[1]);
+        assertEquals("3", array[2]);
+        assertEquals("4", array[3]);
+    }
+
+    @Test
+    public void testShiftAllObject() {
+        String[] array = new String[] {"1", "2", "3", "4"};
+        ArrayUtils.shift(array, 4);
+        assertEquals("1", array[0]);
+        assertEquals("2", array[1]);
+        assertEquals("3", array[2]);
+        assertEquals("4", array[3]);
+        ArrayUtils.shift(array, -4);
+        assertEquals("1", array[0]);
+        assertEquals("2", array[1]);
+        assertEquals("3", array[2]);
+        assertEquals("4", array[3]);
+    }
+    
     //-----------------------------------------------------------------------
     @Test
     public void testIndexOf() {
diff --git a/src/test/java/org/apache/commons/lang3/StringUtilsTest.java b/src/test/java/org/apache/commons/lang3/StringUtilsTest.java
index 5721dbc70a08ca39e19512111bb06393c4b04fcf..f11ec703daadedc63ad32f74ff8edd668be02ff5 100644
--- a/src/test/java/org/apache/commons/lang3/StringUtilsTest.java
+++ b/src/test/java/org/apache/commons/lang3/StringUtilsTest.java
@@ -1353,6 +1353,7 @@ public class StringUtilsTest {
         assertEquals("", StringUtils.repeat("ab", 0));
         assertEquals("", StringUtils.repeat("", 3));
         assertEquals("aaa", StringUtils.repeat("a", 3));
+        assertEquals("", StringUtils.repeat("a", -2));
         assertEquals("ababab", StringUtils.repeat("ab", 3));
         assertEquals("abcabcabc", StringUtils.repeat("abc", 3));
         final String str = StringUtils.repeat("a", 10000);  // bigger than pad limit
@@ -1374,6 +1375,13 @@ public class StringUtilsTest {
         assertEquals("?, ?, ?", StringUtils.repeat("?", ", ", 3));
     }
 
+    @Test
+    public void testRepeat_CharInt() {
+        assertEquals("zzz", StringUtils.repeat('z', 3));
+        assertEquals("", StringUtils.repeat('z', 0));
+        assertEquals("", StringUtils.repeat('z', -2));
+    }
+
     @Test
     public void testChop() {
 
@@ -1617,6 +1625,22 @@ public class StringUtilsTest {
         assertEquals("  abc  ", StringUtils.center("abc", 7, ""));
     }
 
+    //-----------------------------------------------------------------------
+    @Test
+    public void testRotate_StringInt() {
+        assertEquals(null, StringUtils.rotate(null, 1));
+        assertEquals("", StringUtils.rotate("", 1));
+        assertEquals("abcdefg", StringUtils.rotate("abcdefg", 0));
+        assertEquals("fgabcde", StringUtils.rotate("abcdefg", 2));
+        assertEquals("cdefgab", StringUtils.rotate("abcdefg", -2));
+        assertEquals("abcdefg", StringUtils.rotate("abcdefg", 7));
+        assertEquals("abcdefg", StringUtils.rotate("abcdefg", -7));
+        assertEquals("fgabcde", StringUtils.rotate("abcdefg", 9));
+        assertEquals("cdefgab", StringUtils.rotate("abcdefg", -9));
+        assertEquals("efgabcd", StringUtils.rotate("abcdefg", 17));
+        assertEquals("defgabc", StringUtils.rotate("abcdefg", -17));
+    }
+
     //-----------------------------------------------------------------------
     @Test
     public void testReverse_String() {
