diff --git a/tests/test_commands.py b/tests/test_commands.py
index 9189eca2c9758766a866c3e6a5080bd737fcef84..8186add747e1b9fee30ff94c584532c5e6630a1a 100644
--- a/tests/test_commands.py
+++ b/tests/test_commands.py
@@ -68,6 +68,16 @@ def test_auto_shorthelp(runner):
         result.output) is not None
 
 
+def test_no_args_is_help(runner):
+    @click.command(no_args_is_help=True)
+    def cli():
+        pass
+
+    result = runner.invoke(cli, [])
+    assert result.exit_code == 0
+    assert 'Show this message and exit.' in result.output
+
+
 def test_default_maps(runner):
     @click.group()
     def cli():
diff --git a/tests/test_compat.py b/tests/test_compat.py
index a7b05f1ed19aa9521d63727d65184d7502a216cb..abc733b19289d90096d15cd138b619588a45c29e 100644
--- a/tests/test_compat.py
+++ b/tests/test_compat.py
@@ -1,21 +1,21 @@
 import click
 import pytest
+from click._compat import should_strip_ansi, WIN
 
 
-if click.__version__ >= '3.0':
-    def test_legacy_callbacks(runner):
-        def legacy_callback(ctx, value):
-            return value.upper()
+def test_legacy_callbacks(runner):
+    def legacy_callback(ctx, value):
+        return value.upper()
 
-        @click.command()
-        @click.option('--foo', callback=legacy_callback)
-        def cli(foo):
-            click.echo(foo)
+    @click.command()
+    @click.option('--foo', callback=legacy_callback)
+    def cli(foo):
+        click.echo(foo)
 
-        with pytest.warns(Warning, match='Invoked legacy parameter callback'):
-            result = runner.invoke(cli, ['--foo', 'wat'])
-            assert result.exit_code == 0
-            assert 'WAT' in result.output
+    with pytest.warns(Warning, match='Invoked legacy parameter callback'):
+        result = runner.invoke(cli, ['--foo', 'wat'])
+        assert result.exit_code == 0
+        assert 'WAT' in result.output
 
 
 def test_bash_func_name():
@@ -23,3 +23,13 @@ def test_bash_func_name():
     script = get_completion_script('foo-bar baz_blah', '_COMPLETE_VAR', 'bash').strip()
     assert script.startswith('_foo_barbaz_blah_completion()')
     assert '_COMPLETE_VAR=complete $1' in script
+
+
+@pytest.mark.xfail(WIN, reason="Jupyter not tested/supported on Windows")
+def test_is_jupyter_kernel_output():
+    class JupyterKernelFakeStream(object):
+        pass
+
+    # implementation detail, aka cheapskate test
+    JupyterKernelFakeStream.__module__ = "ipykernel.faked"
+    assert not should_strip_ansi(stream=JupyterKernelFakeStream())
diff --git a/tests/test_formatting.py b/tests/test_formatting.py
index 2005f98b5639467895bd81dbe8ec686bfb2967d6..566f3162da58642c8b61c60d2117f9a2069eef45 100644
--- a/tests/test_formatting.py
+++ b/tests/test_formatting.py
@@ -297,3 +297,19 @@ def test_truncating_docstring(runner):
         'Options:',
         '  --help  Show this message and exit.',
     ]
+
+
+def test_global_show_default(runner):
+    @click.command(context_settings=dict(show_default=True))
+    @click.option("-f", "in_file", default="out.txt", help="Output file name")
+    def cli():
+        pass
+
+    result = runner.invoke(cli, ['--help'],)
+    assert result.output.splitlines() == [
+        'Usage: cli [OPTIONS]',
+        '',
+        'Options:',
+        '  -f TEXT  Output file name  [default: out.txt]',
+        '  --help   Show this message and exit.  [default: False]'
+    ]
\ No newline at end of file
diff --git a/tests/test_termui.py b/tests/test_termui.py
index 9e855d3039b300320cdf75fe60347e5f226fd84f..79fb7c6f5d1937a09a8931cc9e6508738ecedb91 100644
--- a/tests/test_termui.py
+++ b/tests/test_termui.py
@@ -1,6 +1,12 @@
+# -*- coding: utf-8 -*-
+import pytest
+
 import click
 import time
 
+import click._termui_impl
+from click._compat import WIN
+
 
 class FakeClock(object):
     def __init__(self):
@@ -13,13 +19,21 @@ class FakeClock(object):
         return self.now
 
 
+def _create_progress(length=10, length_known=True, **kwargs):
+    progress = click.progressbar(tuple(range(length)))
+    for key, value in kwargs.items():
+        setattr(progress, key, value)
+    progress.length_known = length_known
+    return progress
+
+
 def test_progressbar_strip_regression(runner, monkeypatch):
     fake_clock = FakeClock()
     label = '    padded line'
 
     @click.command()
     def cli():
-        with click.progressbar(tuple(range(10)), label=label) as progress:
+        with _create_progress(label=label) as progress:
             for thing in progress:
                 fake_clock.advance_time()
 
@@ -67,7 +81,7 @@ def test_progressbar_hidden(runner, monkeypatch):
 
     @click.command()
     def cli():
-        with click.progressbar(tuple(range(10)), label=label) as progress:
+        with _create_progress(label=label) as progress:
             for thing in progress:
                 fake_clock.advance_time()
 
@@ -76,6 +90,101 @@ def test_progressbar_hidden(runner, monkeypatch):
     assert runner.invoke(cli, []).output == ''
 
 
+@pytest.mark.parametrize('avg, expected', [([], 0.0), ([1, 4], 2.5)])
+def test_progressbar_time_per_iteration(runner, avg, expected):
+    with _create_progress(2, avg=avg) as progress:
+        assert progress.time_per_iteration == expected
+
+
+@pytest.mark.parametrize('finished, expected', [(False, 5), (True, 0)])
+def test_progressbar_eta(runner, finished, expected):
+    with _create_progress(2, finished=finished, avg=[1, 4]) as progress:
+        assert progress.eta == expected
+
+
+@pytest.mark.parametrize('eta, expected',
+                         [(0, '00:00:00'), (30, '00:00:30'), (90, '00:01:30'), (900, '00:15:00'),
+                          (9000, '02:30:00'), (99999999999, '1157407d 09:46:39'), (None, '')])
+def test_progressbar_format_eta(runner, eta, expected):
+    with _create_progress(1, eta_known=eta is not None, avg=[eta]) as progress:
+        assert progress.format_eta() == expected
+
+
+@pytest.mark.parametrize('pos, length', [(0, 5), (-1, 1), (5, 5), (6, 5), (4, 0)])
+def test_progressbar_format_pos(runner, pos, length):
+    with _create_progress(length, length_known=length != 0, pos=pos) as progress:
+        result = progress.format_pos()
+        if progress.length_known:
+            assert result == '%s/%s' % (pos, length)
+        else:
+            assert result == str(pos)
+
+
+@pytest.mark.parametrize('length, finished, pos, avg, expected',
+                         [(8, False, 7, 0, '#######-'),
+                          (0, True, 8, 0, '########'),
+                          (0, False, 8, 0, '--------'),
+                          (0, False, 5, 3, '#-------')
+                          ])
+def test_progressbar_format_bar(runner, length, finished, pos, avg, expected):
+    with _create_progress(length,
+                          length_known=length != 0,
+                          width=8,
+                          pos=pos,
+                          finished=finished,
+                          avg=[avg]) as progress:
+        assert progress.format_bar() == expected
+
+
+@pytest.mark.parametrize('length, length_known, show_percent, show_pos, pos, expected',
+                         [(0, True, True, True, 0, '  [--------]  0/0    0%'),
+                          (0, True, False, True, 0, '  [--------]  0/0'),
+                          (0, True, False, False, 0, '  [--------]'),
+                          (0, False, False, False, 0, '  [--------]'),
+                          (8, True, True, True, 8, '  [########]  8/8  100%')
+                          ])
+def test_progressbar_format_progress_line(runner, length, length_known, show_percent, show_pos, pos, expected):
+    with _create_progress(length,
+                          length_known,
+                          width=8,
+                          show_percent=show_percent,
+                          pos=pos,
+                          show_pos=show_pos) as progress:
+        assert progress.format_progress_line() == expected
+
+
+@pytest.mark.parametrize('test_item', ['test', None])
+def test_progressbar_format_progress_line_with_show_func(runner, test_item):
+
+    def item_show_func(item):
+        return item
+
+    with _create_progress(item_show_func=item_show_func, current_item=test_item) as progress:
+        if test_item:
+            assert progress.format_progress_line().endswith(test_item)
+        else:
+            assert progress.format_progress_line().endswith(progress.format_pct())
+
+
+def test_progressbar_init_exceptions(runner):
+    try:
+        click.progressbar()
+    except TypeError as e:
+        assert str(e) == 'iterable or length is required'
+    else:
+        assert False, 'Expected an exception because unspecified arguments'
+
+
+def test_progressbar_iter_outside_with_exceptions(runner):
+    try:
+        progress = click.progressbar(length=2)
+        iter(progress)
+    except RuntimeError as e:
+        assert str(e) == 'You need to use progress bars in a with block.'
+    else:
+        assert False, 'Expected an exception because of abort-related inputs.'
+
+
 def test_choices_list_in_prompt(runner, monkeypatch):
     @click.command()
     @click.option('-g', type=click.Choice(['none', 'day', 'week', 'month']),
@@ -128,3 +237,39 @@ def test_progressbar_update(runner, monkeypatch):
     assert ' 50%  00:00:02' in lines[1]
     assert ' 75%  00:00:01' in lines[2]
     assert '100%          ' in lines[3]
+
+
+@pytest.mark.parametrize(
+    'key_char', (u'h', u'H', u'é', u'À', u' ', u'字', u'àH', u'àR')
+)
+@pytest.mark.parametrize('echo', [True, False])
+@pytest.mark.skipif(not WIN, reason='Tests user-input using the msvcrt module.')
+def test_getchar_windows(runner, monkeypatch, key_char, echo):
+    monkeypatch.setattr(click._termui_impl.msvcrt, 'getwche', lambda: key_char)
+    monkeypatch.setattr(click._termui_impl.msvcrt, 'getwch', lambda: key_char)
+    monkeypatch.setattr(click.termui, '_getchar', None)
+    assert click.getchar(echo) == key_char
+
+
+@pytest.mark.parametrize('special_key_char, key_char', [(u'\x00', 'a'), (u'\x00', 'b'), (u'\xe0', 'c')])
+@pytest.mark.skipif(not WIN, reason='Tests special character inputs using the msvcrt module.')
+def test_getchar_special_key_windows(runner, monkeypatch, special_key_char, key_char):
+    ordered_inputs = [key_char, special_key_char]
+    monkeypatch.setattr(click._termui_impl.msvcrt, 'getwch', lambda: ordered_inputs.pop())
+    monkeypatch.setattr(click.termui, '_getchar', None)
+    assert click.getchar() == special_key_char + key_char
+
+
+@pytest.mark.parametrize('key_char', [u'\x03', u'\x1a'])
+@pytest.mark.skipif(not WIN, reason='Tests user-input using the msvcrt module.')
+def test_getchar_windows_exceptions(runner, monkeypatch, key_char):
+    monkeypatch.setattr(click._termui_impl.msvcrt, 'getwch', lambda: key_char)
+    monkeypatch.setattr(click.termui, '_getchar', None)
+    try:
+        click.getchar()
+    except KeyboardInterrupt:
+        assert key_char == u'\x03'
+    except EOFError:
+        assert key_char == u'\x1a'
+    else:
+        assert False, 'Expected an exception because of abort-specific inputs.'
