diff --git a/.flake8 b/.flake8
new file mode 100644
index 0000000000000000000000000000000000000000..c53acebcfc5e0ab7f6113f7ab11376959d57bf46
--- /dev/null
+++ b/.flake8
@@ -0,0 +1,25 @@
+[flake8]
+extend-select =
+    # bugbear
+    B
+    # bugbear opinions
+    B9
+    # implicit str concat
+    ISC
+extend-ignore =
+    # slice notation whitespace, invalid
+    E203
+    # line length, handled by bugbear B950
+    E501
+    # bare except, handled by bugbear B001
+    E722
+    # zip with strict=, requires python >= 3.10
+    B905
+    # string formatting opinion, B028 renamed to B907
+    B028
+    B907
+# up to 88 allowed by bugbear B950
+max-line-length = 80
+per-file-ignores =
+    # __init__ exports names
+    src/click/__init__.py: F401
diff --git a/.github/workflows/lock.yaml b/.github/workflows/lock.yaml
index c790fae5cb82c522b0c9142e5c41e0971634ab46..e962fd04109df32b9bc660a3dd1c8909deed0ea0 100644
--- a/.github/workflows/lock.yaml
+++ b/.github/workflows/lock.yaml
@@ -19,7 +19,7 @@ jobs:
   lock:
     runs-on: ubuntu-latest
     steps:
-      - uses: dessant/lock-threads@c1b35aecc5cdb1a34539d14196df55838bb2f836
+      - uses: dessant/lock-threads@be8aa5be94131386884a6da4189effda9b14aa21
         with:
           issue-inactive-days: 14
           pr-inactive-days: 14
diff --git a/.github/workflows/publish.yaml b/.github/workflows/publish.yaml
index 3c8db0e0a2a3964d667989e65fbf8d175e2de2c1..c55ff50b76a2249f084037650a0a3a5927f51b6e 100644
--- a/.github/workflows/publish.yaml
+++ b/.github/workflows/publish.yaml
@@ -9,8 +9,8 @@ jobs:
     outputs:
       hash: ${{ steps.hash.outputs.hash }}
     steps:
-      - uses: actions/checkout@8f4b7f84864484a7bf31766abe9204da3cbe65b3
-      - uses: actions/setup-python@d27e3f3d7c64b4bbf8e4abfb9b63b83e846e0435
+      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9
+      - uses: actions/setup-python@61a6322f88396a6271a6ee3565807d608ecaddd1
         with:
           python-version: '3.x'
           cache: 'pip'
@@ -62,11 +62,10 @@ jobs:
       id-token: write
     steps:
       - uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a
-      # Try uploading to Test PyPI first, in case something fails.
-      - uses: pypa/gh-action-pypi-publish@29930c9cf57955dc1b98162d0d8bc3ec80d9e75c
+      - uses: pypa/gh-action-pypi-publish@f8c70e705ffc13c3b4d1221169b84f12a75d6ca8
         with:
-          repository-url: https://test.pypi.org/legacy/
-          packages-dir: artifact/
-      - uses: pypa/gh-action-pypi-publish@29930c9cf57955dc1b98162d0d8bc3ec80d9e75c
+          repository_url: https://test.pypi.org/legacy/
+          packages_dir: artifact/
+      - uses: pypa/gh-action-pypi-publish@f8c70e705ffc13c3b4d1221169b84f12a75d6ca8
         with:
           packages-dir: artifact/
diff --git a/.github/workflows/tests.yaml b/.github/workflows/tests.yaml
index 0a839517dfa084e8a21670750335e526ebfc1f66..17280318089665c7367d26209423d491c10f87d8 100644
--- a/.github/workflows/tests.yaml
+++ b/.github/workflows/tests.yaml
@@ -35,22 +35,17 @@ jobs:
           - {name: 'PyPy', python: 'pypy-3.10', os: ubuntu-latest, tox: pypy310}
           - {name: Typing, python: '3.11', os: ubuntu-latest, tox: typing}
     steps:
-      - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c
-      - uses: actions/setup-python@5ccb29d8773c3f3f653e1705f474dfaa8a06a912
+      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9
+      - uses: actions/setup-python@61a6322f88396a6271a6ee3565807d608ecaddd1
         with:
           python-version: ${{ matrix.python }}
           cache: 'pip'
           cache-dependency-path: 'requirements/*.txt'
-      - name: update pip
-        run: |
-          pip install -U wheel
-          pip install -U setuptools
-          python -m pip install -U pip
       - name: cache mypy
-        uses: actions/cache@58c146cc91c5b9e778e71775dfe9bf1442ad9a12
+        uses: actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8
         with:
           path: ./.mypy_cache
-          key: mypy|${{ matrix.python }}|${{ hashFiles('setup.cfg') }}
+          key: mypy|${{ matrix.python }}|${{ hashFiles('pyproject.toml') }}
         if: matrix.tox == 'typing'
       - run: pip install tox
       - run: tox run -e ${{ matrix.tox }}
diff --git a/CHANGES.rst b/CHANGES.rst
index ce98025a67074f0e7e6bdb3984ed8987546d0734..957b0a0f6e2678ffa2a1d36a2c5f41ddbfe81bfa 100644
--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -1,5 +1,15 @@
 .. currentmodule:: click
 
+Version 8.2.0
+-------------
+
+Unreleased
+
+-   Use modern packaging metadata with ``pyproject.toml`` instead of ``setup.cfg``.
+    :pr:`326`
+-   Use ``flit_core`` instead of ``setuptools`` as build backend.
+
+
 Version 8.1.4
 -------------
 
diff --git a/CONTRIBUTING.rst b/CONTRIBUTING.rst
index 1ad3bdcbf150c794d219c72d73674cddb901b6d9..0c9d8550511e8f5a501e9db673aa5d189f910fdc 100644
--- a/CONTRIBUTING.rst
+++ b/CONTRIBUTING.rst
@@ -107,12 +107,6 @@ First time setup
 
         > env\Scripts\activate
 
--   Upgrade pip and setuptools.
-
-    .. code-block:: text
-
-        $ python -m pip install --upgrade pip setuptools
-
 -   Install the development dependencies, then install Click in
     editable mode.
 
diff --git a/MANIFEST.in b/MANIFEST.in
deleted file mode 100644
index e5b231d397182c14435d8dfeeb4b40569a7ebf7b..0000000000000000000000000000000000000000
--- a/MANIFEST.in
+++ /dev/null
@@ -1,10 +0,0 @@
-include CHANGES.rst
-include tox.ini
-include requirements/*.txt
-graft artwork
-graft docs
-prune docs/_build
-graft examples
-graft tests
-include src/click/py.typed
-global-exclude *.pyc
diff --git a/docs/commands.rst b/docs/commands.rst
index b70992e51fb15b5430f612bf63ba88be1f16a7f6..62981a51b02fd965949385a7a9803dedaf70d103 100644
--- a/docs/commands.rst
+++ b/docs/commands.rst
@@ -285,11 +285,9 @@ Multi Command Chaining
 .. versionadded:: 3.0
 
 Sometimes it is useful to be allowed to invoke more than one subcommand in
-one go.  For instance if you have installed a setuptools package before
-you might be familiar with the ``setup.py sdist bdist_wheel upload``
-command chain which invokes ``sdist`` before ``bdist_wheel`` before
-``upload``.  Starting with Click 3.0 this is very simple to implement.
-All you have to do is to pass ``chain=True`` to your multicommand:
+one go. For example, ``my-app validate build upload`` would invoke ``validate``,
+then ``build``, then ``upload``. To implement this in Click, pass ``chain=True``
+when creating a group.
 
 .. click:example::
 
diff --git a/docs/entry-points.rst b/docs/entry-points.rst
new file mode 100644
index 0000000000000000000000000000000000000000..2354e01f407a56d6020c0e8396532b835b903e38
--- /dev/null
+++ b/docs/entry-points.rst
@@ -0,0 +1,86 @@
+Packaging Entry Points
+======================
+
+It's recommended to write command line utilities as installable packages with
+entry points instead telling users to run ``python hello.py``.
+
+A distribution package is a ``.whl`` file you install with pip or another Python
+installer. You use a ``pyproject.toml`` file to describe the project and how it
+is built into a package. You might upload this package to PyPI, or distribute it
+to your users in another way.
+
+Python installers create executable scripts that will run a specified Python
+function. These are known as "entry points". The installer knows how to create
+an executable regardless of the operating system, so it will work on Linux,
+Windows, MacOS, etc.
+
+
+Project Files
+-------------
+
+To install your app with an entry point, all you need is the script and a
+``pyproject.toml`` file. Here's an example project directory:
+
+.. code-block:: text
+
+    hello-project/
+        src/
+            hello/
+                __init__.py
+        pyproject.toml
+
+Contents of ``hello.py``:
+
+.. click:example::
+
+    import click
+
+    @click.command()
+    def cli():
+        """Prints a greeting."""
+        click.echo("Hello, World!")
+
+Contents of ``pyproject.toml``:
+
+.. code-block:: toml
+
+    [project]
+    name = "hello"
+    version = "1.0.0"
+    description = "Hello CLI"
+    requires-python = ">=3.11"
+    dependencies = [
+        "click>=8.1",
+    ]
+
+    [project.scripts]
+    hello = "hello:cli"
+
+    [build-system]
+    requires = ["flit_core<4"]
+    build-backend = "flit_core.buildapi"
+
+The magic is in the ``project.scripts`` section. Each line identifies one executable
+script. The first part before the equals sign (``=``) is the name of the script that
+should be generated, the second part is the import path followed by a colon
+(``:``) with the function to call (the Click command).
+
+
+Installation
+------------
+
+When your package is installed, the installer will create an executable script
+based on the configuration. During development, you can install in editable
+mode using the ``-e`` option. Remember to use a virtual environment!
+
+.. code-block:: console
+
+    $ python -m venv .venv
+    $ . .venv/bin/activate
+    $ pip install -e .
+
+Afterwards, your command should be available:
+
+.. click:run::
+
+    invoke(cli, prog_name="hello")
diff --git a/docs/index.rst b/docs/index.rst
index ad965a36bf7c63cac0e4776995013448c1fceed4..61736f4f0bb01e1f1c58f07cae514f29ffeb7039 100644
--- a/docs/index.rst
+++ b/docs/index.rst
@@ -68,7 +68,7 @@ usage patterns.
 
    why
    quickstart
-   setuptools
+   entry-points
    parameters
    options
    arguments
diff --git a/docs/options.rst b/docs/options.rst
index 5c23badba4e62837d0b6a3890180320dff4cac34..2c51ddccf68379a6c7ca1f8810fff89d8e88582a 100644
--- a/docs/options.rst
+++ b/docs/options.rst
@@ -564,7 +564,7 @@ current :class:`Context`, the current :class:`Parameter`, and the value.
 The context provides some useful features such as quitting the
 application and gives access to other already processed parameters.
 
-Here an example for a ``--version`` flag:
+Here's an example for a ``--version`` flag:
 
 .. click:example::
 
diff --git a/docs/quickstart.rst b/docs/quickstart.rst
index 7cd26ca0620ec9792e94f94688a920b7fb16b6e2..61afa0bac7e55cce7f437820c1a189ec74b2505f 100644
--- a/docs/quickstart.rst
+++ b/docs/quickstart.rst
@@ -201,8 +201,7 @@ script can instead be written like this:
     def dropdb():
         click.echo('Dropped the database')
 
-You would then invoke the :class:`Group` in your setuptools entry points or
-other invocations::
+You would then invoke the :class:`Group` in your entry points or other invocations::
 
     if __name__ == '__main__':
         cli()
@@ -251,30 +250,27 @@ What it looks like:
 
     invoke(hello, args=['--help'], prog_name='python hello.py')
 
-.. _switching-to-setuptools:
-
-Switching to Setuptools
------------------------
+Switching to Entry Points
+-------------------------
 
 In the code you wrote so far there is a block at the end of the file which
 looks like this: ``if __name__ == '__main__':``.  This is traditionally
 how a standalone Python file looks like.  With Click you can continue
-doing that, but there are better ways through setuptools.
+doing that, but a better way is to package your app with an entry point.
 
 There are two main (and many more) reasons for this:
 
-The first one is that setuptools automatically generates executable
+The first one is that installers automatically generate executable
 wrappers for Windows so your command line utilities work on Windows too.
 
-The second reason is that setuptools scripts work with virtualenv on Unix
+The second reason is that entry point scripts work with virtualenv on Unix
 without the virtualenv having to be activated.  This is a very useful
 concept which allows you to bundle your scripts with all requirements into
 a virtualenv.
 
 Click is perfectly equipped to work with that and in fact the rest of the
-documentation will assume that you are writing applications through
-setuptools.
+documentation will assume that you are writing applications as distributed
+packages.
 
-I strongly recommend to have a look at the :ref:`setuptools-integration`
-chapter before reading the rest as the examples assume that you will
-be using setuptools.
+Look at the :doc:`entry-points` chapter before reading the rest as the examples
+assume that you will be using entry points.
diff --git a/docs/setuptools.rst b/docs/setuptools.rst
index 7c60c7d3e7327a77eadfa352678ba3fa8b8f05a2..de91b65d3751421b926cc810d8caa49a5bec6bef 100644
--- a/docs/setuptools.rst
+++ b/docs/setuptools.rst
@@ -1,153 +1,6 @@
-.. _setuptools-integration:
+:orphan:
 
 Setuptools Integration
 ======================
 
-When writing command line utilities, it's recommended to write them as
-modules that are distributed with setuptools instead of using Unix
-shebangs.
-
-Why would you want to do that?  There are a bunch of reasons:
-
-1.  One of the problems with the traditional approach is that the first
-    module the Python interpreter loads has an incorrect name.  This might
-    sound like a small issue but it has quite significant implications.
-
-    The first module is not called by its actual name, but the
-    interpreter renames it to ``__main__``.  While that is a perfectly
-    valid name it means that if another piece of code wants to import from
-    that module it will trigger the import a second time under its real
-    name and all of a sudden your code is imported twice.
-
-2.  Not on all platforms are things that easy to execute.  On Linux and OS
-    X you can add a comment to the beginning of the file (``#!/usr/bin/env
-    python``) and your script works like an executable (assuming it has
-    the executable bit set).  This however does not work on Windows.
-    While on Windows you can associate interpreters with file extensions
-    (like having everything ending in ``.py`` execute through the Python
-    interpreter) you will then run into issues if you want to use the
-    script in a virtualenv.
-
-    In fact running a script in a virtualenv is an issue with OS X and
-    Linux as well.  With the traditional approach you need to have the
-    whole virtualenv activated so that the correct Python interpreter is
-    used.  Not very user friendly.
-
-3.  The main trick only works if the script is a Python module.  If your
-    application grows too large and you want to start using a package you
-    will run into issues.
-
-Introduction
-------------
-
-To bundle your script with setuptools, all you need is the script in a
-Python package and a ``setup.py`` file.
-
-Imagine this directory structure:
-
-.. code-block:: text
-
-    yourscript.py
-    setup.py
-
-Contents of ``yourscript.py``:
-
-.. click:example::
-
-    import click
-
-    @click.command()
-    def cli():
-        """Example script."""
-        click.echo('Hello World!')
-
-Contents of ``setup.py``:
-
-.. code-block:: python
-
-    from setuptools import setup
-
-    setup(
-        name='yourscript',
-        version='0.1.0',
-        py_modules=['yourscript'],
-        install_requires=[
-            'Click',
-        ],
-        entry_points={
-            'console_scripts': [
-                'yourscript = yourscript:cli',
-            ],
-        },
-    )
-
-The magic is in the ``entry_points`` parameter.  Below
-``console_scripts``, each line identifies one console script.  The first
-part before the equals sign (``=``) is the name of the script that should
-be generated, the second part is the import path followed by a colon
-(``:``) with the Click command.
-
-That's it.
-
-Testing The Script
-------------------
-
-To test the script, you can make a new virtualenv and then install your
-package:
-
-.. code-block:: console
-
-    $ python3 -m venv .venv
-    $ . .venv/bin/activate
-    $ pip install --editable .
-
-Afterwards, your command should be available:
-
-.. click:run::
-
-    invoke(cli, prog_name='yourscript')
-
-Scripts in Packages
--------------------
-
-If your script is growing and you want to switch over to your script being
-contained in a Python package the changes necessary are minimal.  Let's
-assume your directory structure changed to this:
-
-.. code-block:: text
-
-    project/
-        yourpackage/
-            __init__.py
-            main.py
-            utils.py
-            scripts/
-                __init__.py
-                yourscript.py
-        setup.py
-
-In this case instead of using ``py_modules`` in your ``setup.py`` file you
-can use ``packages`` and the automatic package finding support of
-setuptools.  In addition to that it's also recommended to include other
-package data.
-
-These would be the modified contents of ``setup.py``:
-
-.. code-block:: python
-
-    from setuptools import setup, find_packages
-
-    setup(
-        name='yourpackage',
-        version='0.1.0',
-        packages=find_packages(),
-        include_package_data=True,
-        install_requires=[
-            'Click',
-        ],
-        entry_points={
-            'console_scripts': [
-                'yourscript = yourpackage.scripts.yourscript:cli',
-            ],
-        },
-    )
+Moved to :doc:`entry-points`.
diff --git a/docs/shell-completion.rst b/docs/shell-completion.rst
index ebf73c323e008072d547f4c2694cde583aed4fa5..c49209cb54d073c3a6f9e9126258f8a909ae5fcf 100644
--- a/docs/shell-completion.rst
+++ b/docs/shell-completion.rst
@@ -25,7 +25,7 @@ Enabling Completion
 
 Completion is only available if a script is installed and invoked
 through an entry point, not through the ``python`` command. See
-:doc:`/setuptools`. Once the executable is installed, calling it with
+:doc:`entry-points`. Once the executable is installed, calling it with
 a special environment variable will put Click in completion mode.
 
 In order for completion to be used, the user needs to register a special
diff --git a/examples/README b/examples/README
index 566153fc9cb296fc7b07e3729569c8d4a7a8766d..652dbecb05be8d0b3166faac9b9f741c17a98415 100644
--- a/examples/README
+++ b/examples/README
@@ -3,10 +3,3 @@ Click Examples
   This folder contains various Click examples.  Note that
   all of these are not runnable by themselves but should be
   installed into a virtualenv.
-
-  This is done this way so that scripts also properly work
-  on Windows and in virtualenvs without accidentally executing
-  through the wrong interpreter.
-
-  For more information about this see the documentation:
-    https://click.palletsprojects.com/setuptools/
diff --git a/examples/aliases/pyproject.toml b/examples/aliases/pyproject.toml
new file mode 100644
index 0000000000000000000000000000000000000000..f3a6831f752b85866b00f1ffa41fbf9bd293553b
--- /dev/null
+++ b/examples/aliases/pyproject.toml
@@ -0,0 +1,18 @@
+[project]
+name = "click-example-aliases"
+version = "1.0.0"
+description = "Click aliases example"
+requires-python = ">=3.7"
+dependencies = [
+    "click>=8.1",
+]
+
+[project.scripts]
+aliases = "aliases:cli"
+
+[build-system]
+requires = ["flit_core<4"]
+build-backend = "flit_core.buildapi"
+
+[tool.flit.module]
+name = "aliases"
diff --git a/examples/aliases/setup.py b/examples/aliases/setup.py
deleted file mode 100644
index a3d04e7005fa3df29df789389251dda1b82f6e48..0000000000000000000000000000000000000000
--- a/examples/aliases/setup.py
+++ /dev/null
@@ -1,13 +0,0 @@
-from setuptools import setup
-
-setup(
-    name="click-example-aliases",
-    version="1.0",
-    py_modules=["aliases"],
-    include_package_data=True,
-    install_requires=["click"],
-    entry_points="""
-        [console_scripts]
-        aliases=aliases:cli
-    """,
-)
diff --git a/examples/colors/pyproject.toml b/examples/colors/pyproject.toml
new file mode 100644
index 0000000000000000000000000000000000000000..26446e8714f1158d9464073f3a4f3e59bfb77295
--- /dev/null
+++ b/examples/colors/pyproject.toml
@@ -0,0 +1,18 @@
+[project]
+name = "click-example-colors"
+version = "1.0.0"
+description = "Click colors example"
+requires-python = ">=3.7"
+dependencies = [
+    "click>=8.1",
+]
+
+[project.scripts]
+colors = "colors:cli"
+
+[build-system]
+requires = ["flit_core<4"]
+build-backend = "flit_core.buildapi"
+
+[tool.flit.module]
+name = "colors"
diff --git a/examples/colors/setup.py b/examples/colors/setup.py
deleted file mode 100644
index 3e1a59492615436151aa96d7845ef38749587b0d..0000000000000000000000000000000000000000
--- a/examples/colors/setup.py
+++ /dev/null
@@ -1,13 +0,0 @@
-from setuptools import setup
-
-setup(
-    name="click-example-colors",
-    version="1.0",
-    py_modules=["colors"],
-    include_package_data=True,
-    install_requires=["click"],
-    entry_points="""
-        [console_scripts]
-        colors=colors:cli
-    """,
-)
diff --git a/examples/completion/README b/examples/completion/README
index 372b1d4cacd0f6ad649f282b7eb98137c0f77d18..e2aeca3282c6f44060ecbcdc781aaff6e0c7cb8d 100644
--- a/examples/completion/README
+++ b/examples/completion/README
@@ -26,3 +26,8 @@ For Fish:
     eval (env _COMPLETION_COMPLETE=fish_source completion)
 
 Now press tab (maybe twice) after typing something to see completions.
+
+.. code-block:: python
+
+    $ completion <TAB> <TAB>
+    $ completion gr <TAB> <TAB>
diff --git a/examples/completion/pyproject.toml b/examples/completion/pyproject.toml
new file mode 100644
index 0000000000000000000000000000000000000000..401676cf767c6b569af90214c973195ca5d21ce4
--- /dev/null
+++ b/examples/completion/pyproject.toml
@@ -0,0 +1,18 @@
+[project]
+name = "click-example-completion"
+version = "1.0.0"
+description = "Click completion example"
+requires-python = ">=3.7"
+dependencies = [
+    "click>=8.1",
+]
+
+[project.scripts]
+completion = "completion:cli"
+
+[build-system]
+requires = ["flit_core<4"]
+build-backend = "flit_core.buildapi"
+
+[tool.flit.module]
+name = "completion"
diff --git a/examples/completion/setup.py b/examples/completion/setup.py
deleted file mode 100644
index a78d14087292254112a8894a4d84ae0a0e1d5664..0000000000000000000000000000000000000000
--- a/examples/completion/setup.py
+++ /dev/null
@@ -1,13 +0,0 @@
-from setuptools import setup
-
-setup(
-    name="click-example-completion",
-    version="1.0",
-    py_modules=["completion"],
-    include_package_data=True,
-    install_requires=["click"],
-    entry_points="""
-        [console_scripts]
-        completion=completion:cli
-    """,
-)
diff --git a/examples/complex/pyproject.toml b/examples/complex/pyproject.toml
new file mode 100644
index 0000000000000000000000000000000000000000..2e2c5b0565f937a17ccb27f17eef9df6fec5cf13
--- /dev/null
+++ b/examples/complex/pyproject.toml
@@ -0,0 +1,18 @@
+[project]
+name = "click-example-complex"
+version = "1.0.0"
+description = "Click complex example"
+requires-python = ">=3.7"
+dependencies = [
+    "click>=8.1",
+]
+
+[project.scripts]
+complex = "complex.cli:cli"
+
+[build-system]
+requires = ["flit_core<4"]
+build-backend = "flit_core.buildapi"
+
+[tool.flit.module]
+name = "complex"
diff --git a/examples/complex/setup.py b/examples/complex/setup.py
deleted file mode 100644
index afe97289f4a7d1637967be7ae807efbb95655a2c..0000000000000000000000000000000000000000
--- a/examples/complex/setup.py
+++ /dev/null
@@ -1,13 +0,0 @@
-from setuptools import setup
-
-setup(
-    name="click-example-complex",
-    version="1.0",
-    packages=["complex", "complex.commands"],
-    include_package_data=True,
-    install_requires=["click"],
-    entry_points="""
-        [console_scripts]
-        complex=complex.cli:cli
-    """,
-)
diff --git a/examples/imagepipe/pyproject.toml b/examples/imagepipe/pyproject.toml
new file mode 100644
index 0000000000000000000000000000000000000000..aa6b2dbe011a6bff92a2389f4d2d80dba7ca2d64
--- /dev/null
+++ b/examples/imagepipe/pyproject.toml
@@ -0,0 +1,19 @@
+[project]
+name = "click-example-imagepipe"
+version = "1.0.0"
+description = "Click imagepipe example"
+requires-python = ">=3.7"
+dependencies = [
+    "click>=8.1",
+    "pillow",
+]
+
+[project.scripts]
+imagepipe = "imagepipe:cli"
+
+[build-system]
+requires = ["flit_core<4"]
+build-backend = "flit_core.buildapi"
+
+[tool.flit.module]
+name = "imagepipe"
diff --git a/examples/imagepipe/setup.py b/examples/imagepipe/setup.py
deleted file mode 100644
index c42b5ff60547e27fad687cb2551aeac991d0b0a8..0000000000000000000000000000000000000000
--- a/examples/imagepipe/setup.py
+++ /dev/null
@@ -1,13 +0,0 @@
-from setuptools import setup
-
-setup(
-    name="click-example-imagepipe",
-    version="1.0",
-    py_modules=["imagepipe"],
-    include_package_data=True,
-    install_requires=["click", "pillow"],
-    entry_points="""
-        [console_scripts]
-        imagepipe=imagepipe:cli
-    """,
-)
diff --git a/examples/inout/pyproject.toml b/examples/inout/pyproject.toml
new file mode 100644
index 0000000000000000000000000000000000000000..6ad3ed9da843442e89fb5f240f796d24c5895963
--- /dev/null
+++ b/examples/inout/pyproject.toml
@@ -0,0 +1,18 @@
+[project]
+name = "click-example-inout"
+version = "1.0.0"
+description = "Click inout example"
+requires-python = ">=3.7"
+dependencies = [
+    "click>=8.1",
+]
+
+[project.scripts]
+inout = "inout:cli"
+
+[build-system]
+requires = ["flit_core<4"]
+build-backend = "flit_core.buildapi"
+
+[tool.flit.module]
+name = "inout"
diff --git a/examples/inout/setup.py b/examples/inout/setup.py
deleted file mode 100644
index ff673e3b93337862414d946168b3255a72a47514..0000000000000000000000000000000000000000
--- a/examples/inout/setup.py
+++ /dev/null
@@ -1,13 +0,0 @@
-from setuptools import setup
-
-setup(
-    name="click-example-inout",
-    version="0.1",
-    py_modules=["inout"],
-    include_package_data=True,
-    install_requires=["click"],
-    entry_points="""
-        [console_scripts]
-        inout=inout:cli
-    """,
-)
diff --git a/examples/naval/pyproject.toml b/examples/naval/pyproject.toml
new file mode 100644
index 0000000000000000000000000000000000000000..6d233b87eb9f763b89d0985ae5024b79e4f18d20
--- /dev/null
+++ b/examples/naval/pyproject.toml
@@ -0,0 +1,18 @@
+[project]
+name = "click-example-naval"
+version = "1.0.0"
+description = "Click naval example"
+requires-python = ">=3.7"
+dependencies = [
+    "click>=8.1",
+]
+
+[project.scripts]
+naval = "naval:cli"
+
+[build-system]
+requires = ["flit_core<4"]
+build-backend = "flit_core.buildapi"
+
+[tool.flit.module]
+name = "naval"
diff --git a/examples/naval/setup.py b/examples/naval/setup.py
deleted file mode 100644
index 37b39f5d249ca4fa71a619e9f30619e39a3d4e82..0000000000000000000000000000000000000000
--- a/examples/naval/setup.py
+++ /dev/null
@@ -1,13 +0,0 @@
-from setuptools import setup
-
-setup(
-    name="click-example-naval",
-    version="2.0",
-    py_modules=["naval"],
-    include_package_data=True,
-    install_requires=["click"],
-    entry_points="""
-        [console_scripts]
-        naval=naval:cli
-    """,
-)
diff --git a/examples/repo/pyproject.toml b/examples/repo/pyproject.toml
new file mode 100644
index 0000000000000000000000000000000000000000..30801bd20f1f17280a4fe3fc04ff88242183e006
--- /dev/null
+++ b/examples/repo/pyproject.toml
@@ -0,0 +1,18 @@
+[project]
+name = "click-example-repo"
+version = "1.0.0"
+description = "Click repo example"
+requires-python = ">=3.7"
+dependencies = [
+    "click>=8.1",
+]
+
+[project.scripts]
+repo = "repo:cli"
+
+[build-system]
+requires = ["flit_core<4"]
+build-backend = "flit_core.buildapi"
+
+[tool.flit.module]
+name = "repo"
diff --git a/examples/repo/setup.py b/examples/repo/setup.py
deleted file mode 100644
index 3028020d8257ed4a94b8581714d6f19a5b6f9bea..0000000000000000000000000000000000000000
--- a/examples/repo/setup.py
+++ /dev/null
@@ -1,13 +0,0 @@
-from setuptools import setup
-
-setup(
-    name="click-example-repo",
-    version="0.1",
-    py_modules=["repo"],
-    include_package_data=True,
-    install_requires=["click"],
-    entry_points="""
-        [console_scripts]
-        repo=repo:cli
-    """,
-)
diff --git a/examples/termui/pyproject.toml b/examples/termui/pyproject.toml
new file mode 100644
index 0000000000000000000000000000000000000000..b6cab67d925e69a1e2eb3a5ac2eb5ec04426a7d7
--- /dev/null
+++ b/examples/termui/pyproject.toml
@@ -0,0 +1,18 @@
+[project]
+name = "click-example-termui"
+version = "1.0.0"
+description = "Click termui example"
+requires-python = ">=3.7"
+dependencies = [
+    "click>=8.1",
+]
+
+[project.scripts]
+termui = "termui:cli"
+
+[build-system]
+requires = ["flit_core<4"]
+build-backend = "flit_core.buildapi"
+
+[tool.flit.module]
+name = "termui"
diff --git a/examples/termui/setup.py b/examples/termui/setup.py
deleted file mode 100644
index c1ac109fef96defa5c4a5fcef664fb98bdfe323b..0000000000000000000000000000000000000000
--- a/examples/termui/setup.py
+++ /dev/null
@@ -1,13 +0,0 @@
-from setuptools import setup
-
-setup(
-    name="click-example-termui",
-    version="1.0",
-    py_modules=["termui"],
-    include_package_data=True,
-    install_requires=["click"],
-    entry_points="""
-        [console_scripts]
-        termui=termui:cli
-    """,
-)
diff --git a/examples/validation/pyproject.toml b/examples/validation/pyproject.toml
new file mode 100644
index 0000000000000000000000000000000000000000..77701c1a4bc9c95c6d8d0e84de7f91d191ca34f0
--- /dev/null
+++ b/examples/validation/pyproject.toml
@@ -0,0 +1,18 @@
+[project]
+name = "click-example-validation"
+version = "1.0.0"
+description = "Click validation example"
+requires-python = ">=3.7"
+dependencies = [
+    "click>=8.1",
+]
+
+[project.scripts]
+validation = "validation:cli"
+
+[build-system]
+requires = ["flit_core<4"]
+build-backend = "flit_core.buildapi"
+
+[tool.flit.module]
+name = "validation"
diff --git a/examples/validation/setup.py b/examples/validation/setup.py
deleted file mode 100644
index b7698f61bfa11a101e233aca6e574be7eab1daeb..0000000000000000000000000000000000000000
--- a/examples/validation/setup.py
+++ /dev/null
@@ -1,13 +0,0 @@
-from setuptools import setup
-
-setup(
-    name="click-example-validation",
-    version="1.0",
-    py_modules=["validation"],
-    include_package_data=True,
-    install_requires=["click"],
-    entry_points="""
-        [console_scripts]
-        validation=validation:cli
-    """,
-)
diff --git a/pyproject.toml b/pyproject.toml
new file mode 100644
index 0000000000000000000000000000000000000000..edc47b4be72b0636687c591510253fa0bd85a2d2
--- /dev/null
+++ b/pyproject.toml
@@ -0,0 +1,74 @@
+[project]
+name = "click"
+description = "Composable command line interface toolkit"
+readme = "README.rst"
+license = {file = "LICENSE.rst"}
+maintainers = [{name = "Pallets", email = "contact@palletsprojects.com"}]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "License :: OSI Approved :: BSD License",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python",
+]
+requires-python = ">=3.7"
+dependencies = [
+    "colorama; platform_system == 'Windows'",
+    "importlib-metadata; python_version < '3.8'",
+]
+dynamic = ["version"]
+
+[project.urls]
+Donate = "https://palletsprojects.com/donate"
+Documentation = "https://click.palletsprojects.com/"
+Changes = "https://click.palletsprojects.com/changes/"
+"Source Code" = "https://github.com/pallets/click/"
+"Issue Tracker" = "https://github.com/pallets/click/issues/"
+Chat = "https://discord.gg/pallets"
+
+[build-system]
+requires = ["flit_core<4"]
+build-backend = "flit_core.buildapi"
+
+[tool.flit.module]
+name = "click"
+
+[tool.flit.sdist]
+include = [
+    "docs/",
+    "examples/",
+    "requirements/",
+    "tests/",
+    "CHANGES.rst",
+    "tox.ini",
+]
+exclude = [
+    "docs/_build/",
+]
+
+[tool.pytest.ini_options]
+testpaths = ["tests"]
+filterwarnings = ["error"]
+
+[tool.coverage.run]
+branch = true
+source = ["click", "tests"]
+
+[tool.coverage.paths]
+source = ["src", "*/site-packages"]
+
+[tool.mypy]
+python_version = "3.7"
+files = ["src/click"]
+show_error_codes = true
+pretty = true
+strict = true
+local_partial_types = true
+warn_unreachable = true
+
+[[tool.mypy.overrides]]
+module = [
+    "colorama.*",
+    "importlib_metadata.*"
+]
+ignore_missing_imports = true
diff --git a/setup.cfg b/setup.cfg
deleted file mode 100644
index 7446f699b9f5141c61d8baaff41968d0b64bfabb..0000000000000000000000000000000000000000
--- a/setup.cfg
+++ /dev/null
@@ -1,104 +0,0 @@
-[metadata]
-name = click
-version = attr: click.__version__
-url = https://palletsprojects.com/p/click/
-project_urls =
-    Donate = https://palletsprojects.com/donate
-    Documentation = https://click.palletsprojects.com/
-    Changes = https://click.palletsprojects.com/changes/
-    Source Code = https://github.com/pallets/click/
-    Issue Tracker = https://github.com/pallets/click/issues/
-    Chat = https://discord.gg/pallets
-license = BSD-3-Clause
-license_files = LICENSE.rst
-maintainer = Pallets
-maintainer_email = contact@palletsprojects.com
-description = Composable command line interface toolkit
-long_description = file: README.rst
-long_description_content_type = text/x-rst
-classifiers =
-    Development Status :: 5 - Production/Stable
-    Intended Audience :: Developers
-    License :: OSI Approved :: BSD License
-    Operating System :: OS Independent
-    Programming Language :: Python
-
-[options]
-packages = find:
-package_dir = = src
-include_package_data = True
-python_requires = >= 3.7
-# Dependencies are in setup.py for GitHub's dependency graph.
-
-[options.packages.find]
-where = src
-
-[tool:pytest]
-testpaths = tests
-filterwarnings =
-    error
-
-[coverage:run]
-branch = True
-source =
-    click
-    tests
-
-[coverage:paths]
-source =
-    src
-    */site-packages
-
-[flake8]
-# B = bugbear
-# E = pycodestyle errors
-# F = flake8 pyflakes
-# W = pycodestyle warnings
-# B9 = bugbear opinions
-# ISC = implicit str concat
-select = B, E, F, W, B9, ISC
-ignore =
-    # slice notation whitespace, invalid
-    E203
-    # line length, handled by bugbear B950
-    E501
-    # bare except, handled by bugbear B001
-    E722
-    # bin op line break, invalid
-    W503
-    # zip with strict=, requires python >= 3.10
-    B905
-    # string formatting opinion, B028 renamed to B907
-    B028
-    B907
-# up to 88 allowed by bugbear B950
-max-line-length = 80
-per-file-ignores =
-    # __init__ exports names
-    src/click/__init__.py: F401
-
-[mypy]
-files = src/click
-python_version = 3.7
-show_error_codes = True
-disallow_subclassing_any = True
-disallow_untyped_calls = True
-disallow_untyped_defs = True
-disallow_incomplete_defs = True
-disallow_any_generics = True
-check_untyped_defs = True
-no_implicit_optional = True
-local_partial_types = True
-no_implicit_reexport = True
-strict_equality = True
-warn_redundant_casts = True
-warn_unused_configs = True
-warn_unused_ignores = True
-warn_return_any = True
-warn_unreachable = True
-
-[mypy-colorama.*]
-ignore_missing_imports = True
-
-[mypy-importlib_metadata.*]
-ignore_missing_imports = True
diff --git a/setup.py b/setup.py
deleted file mode 100644
index 0a74d4144abfd4ecfe747bd1ea0780c4979fd07f..0000000000000000000000000000000000000000
--- a/setup.py
+++ /dev/null
@@ -1,9 +0,0 @@
-from setuptools import setup
-
-setup(
-    name="click",
-    install_requires=[
-        "colorama; platform_system == 'Windows'",
-        "importlib-metadata; python_version < '3.8'",
-    ],
-)
diff --git a/src/click/__init__.py b/src/click/__init__.py
index 4704758b2a7184dc887e998c7e946682ac3a593b..a6e979972980c29d5feddb1fd9b621979bd1487b 100644
--- a/src/click/__init__.py
+++ b/src/click/__init__.py
@@ -70,4 +70,4 @@ from .utils import get_binary_stream as get_binary_stream
 from .utils import get_text_stream as get_text_stream
 from .utils import open_file as open_file
 
-__version__ = "8.1.4"
+__version__ = "8.2.0.dev0"
