diff --git a/tests/test_utils.py b/tests/test_utils.py
index 12709a1d8b75f6fe8c0270daf3fba7b710505807..5e629b5a95537bf097366e1c0489db30d2791b99 100644
--- a/tests/test_utils.py
+++ b/tests/test_utils.py
@@ -464,7 +464,7 @@ def test_expand_args(monkeypatch):
     assert user in click.utils._expand_args(["~"])
     monkeypatch.setenv("CLICK_TEST", "hello")
     assert "hello" in click.utils._expand_args(["$CLICK_TEST"])
-    assert "setup.cfg" in click.utils._expand_args(["*.cfg"])
+    assert "pyproject.toml" in click.utils._expand_args(["*.toml"])
     assert os.path.join("docs", "conf.py") in click.utils._expand_args(["**/conf.py"])
     assert "*.not-found" in click.utils._expand_args(["*.not-found"])
     # a bad glob pattern, such as a pytest identifier, should return itself
