diff --git a/src/click/shell_completion.py b/src/click/shell_completion.py
index cdb58222c5b0fd5fd6c4c328e940b29caab3a2fb..b29e442fe23d13bdf6e6e63925e067463e06aa8a 100644
--- a/src/click/shell_completion.py
+++ b/src/click/shell_completion.py
@@ -532,6 +532,7 @@ def _is_incomplete_option(ctx: Context, args: list[str], param: Parameter) -> bo
 
         if _start_of_option(ctx, arg):
             last_option = arg
+            break
 
     return last_option is not None and last_option in param.opts
 
diff --git a/tests/test_shell_completion.py b/tests/test_shell_completion.py
index 3511f0743ab7100aa565b4f37d4f1ae75b9c7501..830dc92da7c754864b082fcc24f636dcdc93ee21 100644
--- a/tests/test_shell_completion.py
+++ b/tests/test_shell_completion.py
@@ -188,6 +188,19 @@ def test_option_flag():
     assert _get_words(cli, ["--on"], "a") == ["a1", "a2"]
 
 
+def test_flag_option_with_nargs_option():
+    cli = Command(
+        "cli",
+        add_help_option=False,
+        params=[
+            Argument(["a"], type=Choice(["a1", "a2", "b"])),
+            Option(["--flag"], is_flag=True),
+            Option(["-c"], type=Choice(["p", "q"]), nargs=2),
+        ],
+    )
+    assert _get_words(cli, ["a1", "--flag", "-c"], "") == ["p", "q"]
+
+
 def test_option_custom():
     def custom(ctx, param, incomplete):
         return [incomplete.upper()]
