diff --git a/CHANGES.rst b/CHANGES.rst
index 6021a5640bf462615996f29dfb15d091f135566a..d6baa2b095f9e1bb0f38869c4651bcc59a9ca4c7 100644
--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -15,6 +15,7 @@ Unreleased
     screen. Refs :issue:`2911` :pr:`3004`
 -   Fix completion for the Z shell (``zsh``) for completion items containing
     colons. :issue:`2703` :pr:`2846`
+-   Don't include envvar in error hint when not configured. :issue:`2971` :pr:`2972`
 -   Fix a rare race in ``click.testing.StreamMixer``'s finalization that manifested
     as a ``ValueError`` on close in a multi-threaded test session.
     :issue:`2993` :pr:`2991`
diff --git a/src/click/core.py b/src/click/core.py
index 4bebb77891f3b425196f05b4c5632eb3640aaaae..583cc895d9345c93bd0be6ea5fb6a6d43d0d4dc7 100644
--- a/src/click/core.py
+++ b/src/click/core.py
@@ -2682,7 +2682,7 @@ class Option(Parameter):
 
     def get_error_hint(self, ctx: Context) -> str:
         result = super().get_error_hint(ctx)
-        if self.show_envvar:
+        if self.show_envvar and self.envvar is not None:
             result += f" (env var: '{self.envvar}')"
         return result
 
diff --git a/tests/test_options.py b/tests/test_options.py
index 1e7826d82a452283781f985a05d8274007f89d55..c5f2d49c7c00f9ac122c6b4cc2f3d1fac9ab800b 100644
--- a/tests/test_options.py
+++ b/tests/test_options.py
@@ -677,6 +677,13 @@ def test_missing_envvar(runner):
     assert result.exit_code == 2
     assert "Error: Missing option '--foo' (env var: 'bar')." in result.output
 
+    cli = click.Command(
+        "cli", params=[click.Option(["--foo"], show_envvar=True, required=True)]
+    )
+    result = runner.invoke(cli)
+    assert result.exit_code == 2
+    assert "Error: Missing option '--foo'." in result.output
+
 
 def test_case_insensitive_choice(runner):
     @click.command()
