diff --git a/tests/test_termui.py b/tests/test_termui.py
index 7cfa939949f9d4ba1fe1f685acccf9f2182ff9f0..2af7ea18f4ac4d480a31db40266f463398fd2817 100644
--- a/tests/test_termui.py
+++ b/tests/test_termui.py
@@ -447,3 +447,15 @@ def test_confirmation_prompt(runner, prompt, input, default, expect):
 
     if prompt == "Confirm Password":
         assert "Confirm Password: " in result.output
+
+
+def test_false_show_default_cause_no_default_display_in_prompt(runner):
+    @click.command()
+    @click.option("--arg1", show_default=False, prompt=True, default="my-default-value")
+    def cmd(arg1):
+        pass
+
+    # Confirm that the default value is not included in the output when `show_default`
+    # is False
+    result = runner.invoke(cmd, input="my-input", standalone_mode=False)
+    assert "my-default-value" not in result.output
