diff --git a/src/click/shell_completion.py b/src/click/shell_completion.py
index cdb58222c5b0fd5fd6c4c328e940b29caab3a2fb..6c39d5eb3889000b499d37262cc5f7de8b7a443e 100644
--- a/src/click/shell_completion.py
+++ b/src/click/shell_completion.py
@@ -567,8 +567,8 @@ def _resolve_context(
                     with cmd.make_context(
                         name, args, parent=ctx, resilient_parsing=True
                     ) as sub_ctx:
-                        args = ctx._protected_args + ctx.args
                         ctx = sub_ctx
+                        args = ctx._protected_args + ctx.args
                 else:
                     sub_ctx = ctx
 
@@ -586,8 +586,8 @@ def _resolve_context(
                             allow_interspersed_args=False,
                             resilient_parsing=True,
                         ) as sub_sub_ctx:
-                            args = sub_ctx.args
                             sub_ctx = sub_sub_ctx
+                            args = sub_ctx.args
 
                     ctx = sub_ctx
                     args = [*sub_ctx._protected_args, *sub_ctx.args]
diff --git a/tests/test_shell_completion.py b/tests/test_shell_completion.py
index 3511f0743ab7100aa565b4f37d4f1ae75b9c7501..fb39486dcc23d441d74f8244657574a5496282c6 100644
--- a/tests/test_shell_completion.py
+++ b/tests/test_shell_completion.py
@@ -40,6 +40,36 @@ def test_group():
     assert _get_words(cli, [], "-") == ["-a", "--help"]
 
 
+@pytest.mark.parametrize(
+    ("args", "word", "expect"),
+    [
+        ([], "", ["get"]),
+        (["get"], "", ["full"]),
+        (["get", "full"], "", ["data"]),
+        (["get", "full"], "-", ["--verbose", "--help"]),
+        (["get", "full", "data"], "", []),
+        (["get", "full", "data"], "-", ["-a", "--help"]),
+    ],
+)
+def test_nested_group(args: list[str], word: str, expect: list[str]) -> None:
+    cli = Group(
+        "cli",
+        commands=[
+            Group(
+                "get",
+                commands=[
+                    Group(
+                        "full",
+                        params=[Option(["--verbose"])],
+                        commands=[Command("data", params=[Option(["-a"])])],
+                    )
+                ],
+            )
+        ],
+    )
+    assert _get_words(cli, args, word) == expect
+
+
 def test_group_command_same_option():
     cli = Group(
         "cli", params=[Option(["-a"])], commands=[Command("x", params=[Option(["-a"])])]
