diff --git a/src/click/core.py b/src/click/core.py
index 4745b533e82037617456a1238143ff2898c66cf4..8aa4b672c1bba8f373c4823695f7d7c98d9e4884 100644
--- a/src/click/core.py
+++ b/src/click/core.py
@@ -2675,7 +2675,7 @@ class Option(Parameter):
 
     def get_error_hint(self, ctx: Context) -> str:
         result = super().get_error_hint(ctx)
-        if self.show_envvar:
+        if self.show_envvar and self.envvar is not None:
             result += f" (env var: '{self.envvar}')"
         return result
 
diff --git a/tests/test_arguments.py b/tests/test_arguments.py
index 50db8e606f808706fd750baf7ea8564ba5a5b926..8b2acbb0f6bf8c783960d3e0fb257cf7c4dc5059 100644
--- a/tests/test_arguments.py
+++ b/tests/test_arguments.py
@@ -63,6 +63,15 @@ def test_nargs_tup_composite(runner, opts):
     assert result.output.splitlines() == ["name=peter id=1"]
 
 
+def test_nargs_mismatch_with_tuple_type():
+    with pytest.raises(ValueError, match="nargs.*must be 2.*but it was 3"):
+
+        @click.command()
+        @click.argument("test", type=(str, int), nargs=3)
+        def cli(_):
+            pass
+
+
 def test_nargs_err(runner):
     @click.command()
     @click.argument("x")
diff --git a/tests/test_options.py b/tests/test_options.py
index dd87bc8cdd966a980846a0de8ff21acc3ded7970..bb92d6197bb4972ffd5ae70daef8366af30a6d0e 100644
--- a/tests/test_options.py
+++ b/tests/test_options.py
@@ -592,6 +592,13 @@ def test_missing_envvar(runner):
     assert result.exit_code == 2
     assert "Error: Missing option '--foo' (env var: 'bar')." in result.output
 
+    cli = click.Command(
+        "cli", params=[click.Option(["--foo"], show_envvar=True, required=True)]
+    )
+    result = runner.invoke(cli)
+    assert result.exit_code == 2
+    assert "Error: Missing option '--foo'." in result.output
+
 
 def test_case_insensitive_choice(runner):
     @click.command()
@@ -1100,3 +1107,33 @@ def test_duplicate_names_warning(runner, opts_one, opts_two):
 
     with pytest.warns(UserWarning):
         runner.invoke(cli, [])
+
+
+@pytest.mark.parametrize(
+    ("multiple", "nargs", "default", "expected_message"),
+    [
+        (
+            False,  # multiple
+            2,  # nargs
+            42,  # default (not a list)
+            "'default' must be a list when 'nargs' != 1.",
+        ),
+        (
+            True,  # multiple
+            2,  # nargs
+            [
+                "test string which is not a list in the list"
+            ],  # default (list of non-lists)
+            "'default' must be a list of lists when 'multiple' is true"
+            " and 'nargs' != 1.",
+        ),
+    ],
+)
+def test_default_type_error_raises(multiple, nargs, default, expected_message):
+    with pytest.raises(ValueError, match=expected_message):
+        click.Option(
+            ["--test"],
+            multiple=multiple,
+            nargs=nargs,
+            default=default,
+        )
