diff --git a/src/click/types.py b/src/click/types.py
index e71c1c21e4c2e1f954ceeed228507478f138993a..73ebd3f5bbfa54b8205db44a92ce16ad68f8ac13 100644
--- a/src/click/types.py
+++ b/src/click/types.py
@@ -387,8 +387,7 @@ class Choice(ParamType, t.Generic[ParamTypeValue]):
         """
         from click.shell_completion import CompletionItem
 
-        str_choices = map(str, self.choices)
-
+        str_choices = [self.normalize_choice(choice, ctx) for choice in self.choices]
         if self.case_sensitive:
             matched = (c for c in str_choices if c.startswith(incomplete))
         else:
diff --git a/tests/test_shell_completion.py b/tests/test_shell_completion.py
index 20cff238f7c72d4999b20205c1fa0244f9934d70..8993978258542593062af62808972621ee775c7e 100644
--- a/tests/test_shell_completion.py
+++ b/tests/test_shell_completion.py
@@ -457,7 +457,8 @@ def test_context_settings(runner):
     assert result.output == "plain,a\nplain,b\n"
 
 
-@pytest.mark.parametrize(("value", "expect"), [(False, ["Au", "al"]), (True, ["al"])])
+# case_sensitive=False normalizes values to lowercase, matching remains case insensitive
+@pytest.mark.parametrize(("value", "expect"), [(False, ["au", "al"]), (True, ["al"])])
 def test_choice_case_sensitive(value, expect):
     cli = Command(
         "cli",
