diff --git a/click/testing.py b/click/testing.py
index f13d4bf201b8557a537f282401e630e08ae4a21a..bfa67fb7d50b03b16e9b9d93caca386af381d7fd 100644
--- a/click/testing.py
+++ b/click/testing.py
@@ -318,8 +318,12 @@ class CliRunner(object):
                 args = shlex.split(args)
 
             try:
-                cli.main(args=args or (),
-                         prog_name=self.get_default_prog_name(cli), **extra)
+                prog_name = extra.pop("prog_name")
+            except KeyError:
+                prog_name = self.get_default_prog_name(cli)
+
+            try:
+                cli.main(args=args or (), prog_name=prog_name, **extra)
             except SystemExit as e:
                 exc_info = sys.exc_info()
                 exit_code = e.code
diff --git a/tests/test_testing.py b/tests/test_testing.py
index a74c5de8ff87b990049c80ec34ecfc0f4c2c1e1c..12f57de3926b5ee7cf7d91b2b0cdfe30b1560641 100644
--- a/tests/test_testing.py
+++ b/tests/test_testing.py
@@ -246,3 +246,14 @@ def test_args(args, expected_output):
     result = runner.invoke(cli_args, args=args)
     assert result.exit_code == 0
     assert result.output == expected_output
+
+
+def test_setting_prog_name_in_extra():
+    @click.command()
+    def cli():
+        click.echo("ok")
+
+    runner = CliRunner()
+    result = runner.invoke(cli, prog_name="foobar")
+    assert not result.exception
+    assert result.output == 'ok\n'
