diff --git a/click/types.py b/click/types.py
index 879968248a7fac2a76a8b3254b2c5551ee787185..f59400529e4b93b5c869e631e7934644a34f28fa 100644
--- a/click/types.py
+++ b/click/types.py
@@ -277,7 +277,7 @@ class BoolParamType(ParamType):
         if isinstance(value, bool):
             return bool(value)
         value = value.lower()
-        if value in ('true', 't,' '1', 'yes', 'y'):
+        if value in ('true', 't', '1', 'yes', 'y'):
             return True
         elif value in ('false', 'f', '0', 'no', 'n'):
             return False
diff --git a/tests/test_basic.py b/tests/test_basic.py
index 94e0aaea0259b69177e78cbfcb30e5ed907c2b46..8ba251fa1a889c9854bad6859cbd0f33bc18c5da 100644
--- a/tests/test_basic.py
+++ b/tests/test_basic.py
@@ -187,7 +187,7 @@ def test_boolean_conversion(runner):
         def cli(flag):
             click.echo(flag)
 
-        for value in 'true', 't,' '1', 'yes', 'y':
+        for value in 'true', 't', '1', 'yes', 'y':
             result = runner.invoke(cli, ['--flag', value])
             assert not result.exception
             assert result.output == 'True\n'
