diff --git a/tests/test_options.py b/tests/test_options.py
index b209e0104dda2e5fc5e6e139dd30d3699cc40202..eefc260e24b670c552a38f70f626b2ef06992b02 100644
--- a/tests/test_options.py
+++ b/tests/test_options.py
@@ -41,6 +41,9 @@ def test_counting(runner):
     assert not result.exception
     assert result.output == 'verbosity=0\n'
 
+    result = runner.invoke(cli, ['--help'])
+    assert re.search('-v\s+Verbosity', result.output) is not None
+
 
 @pytest.mark.parametrize('unknown_flag', ['--foo', '-f'])
 def test_unknown_options(runner, unknown_flag):
