diff --git a/tests/test_arguments.py b/tests/test_arguments.py
index 0b510c00b8aeed45f6af134b17148f08ce44916b..c5fee42423f2eb8fc6fa124f466213d8e0a00c23 100644
--- a/tests/test_arguments.py
+++ b/tests/test_arguments.py
@@ -1,11 +1,8 @@
-# -*- coding: utf-8 -*-
 import sys
 
 import pytest
 
 import click
-from click._compat import PY2
-from click._compat import text_type
 
 
 def test_nargs_star(runner):
@@ -13,8 +10,8 @@ def test_nargs_star(runner):
     @click.argument("src", nargs=-1)
     @click.argument("dst")
     def copy(src, dst):
-        click.echo("src={}".format("|".join(src)))
-        click.echo("dst={}".format(dst))
+        click.echo(f"src={'|'.join(src)}")
+        click.echo(f"dst={dst}")
 
     result = runner.invoke(copy, ["foo.txt", "bar.txt", "dir"])
     assert not result.exception
@@ -35,8 +32,9 @@ def test_nargs_tup(runner):
     @click.argument("name", nargs=1)
     @click.argument("point", nargs=2, type=click.INT)
     def copy(name, point):
-        click.echo("name={}".format(name))
-        click.echo("point={0[0]}/{0[1]}".format(point))
+        click.echo(f"name={name}")
+        x, y = point
+        click.echo(f"point={x}/{y}")
 
     result = runner.invoke(copy, ["peter", "1", "2"])
     assert not result.exception
@@ -56,7 +54,8 @@ def test_nargs_tup_composite(runner):
         @click.command()
         @click.argument("item", **opts)
         def copy(item):
-            click.echo("name={0[0]} id={0[1]:d}".format(item))
+            name, id = item
+            click.echo(f"name={name} id={id:d}")
 
         result = runner.invoke(copy, ["peter", "1"])
         assert not result.exception
@@ -83,22 +82,17 @@ def test_bytes_args(runner, monkeypatch):
     @click.argument("arg")
     def from_bytes(arg):
         assert isinstance(
-            arg, text_type
+            arg, str
         ), "UTF-8 encoded argument should be implicitly converted to Unicode"
 
     # Simulate empty locale environment variables
-    if PY2:
-        monkeypatch.setattr(sys.stdin, "encoding", "ANSI_X3.4-1968")
-        monkeypatch.setattr(sys, "getfilesystemencoding", lambda: "ANSI_X3.4-1968")
-        monkeypatch.setattr(sys, "getdefaultencoding", lambda: "ascii")
-    else:
-        monkeypatch.setattr(sys.stdin, "encoding", "utf-8")
-        monkeypatch.setattr(sys, "getfilesystemencoding", lambda: "utf-8")
-        monkeypatch.setattr(sys, "getdefaultencoding", lambda: "utf-8")
+    monkeypatch.setattr(sys.stdin, "encoding", "utf-8")
+    monkeypatch.setattr(sys, "getfilesystemencoding", lambda: "utf-8")
+    monkeypatch.setattr(sys, "getdefaultencoding", lambda: "utf-8")
 
     runner.invoke(
         from_bytes,
-        [u"Something outside of ASCII range: 林".encode("UTF-8")],
+        ["Something outside of ASCII range: 林".encode()],
         catch_exceptions=False,
     )
 
@@ -195,7 +189,7 @@ def test_empty_nargs(runner):
     @click.command()
     @click.argument("arg", nargs=-1)
     def cmd(arg):
-        click.echo("arg:{}".format("|".join(arg)))
+        click.echo(f"arg:{'|'.join(arg)}")
 
     result = runner.invoke(cmd, [])
     assert result.exit_code == 0
@@ -204,7 +198,7 @@ def test_empty_nargs(runner):
     @click.command()
     @click.argument("arg", nargs=-1, required=True)
     def cmd2(arg):
-        click.echo("arg:{}".format("|".join(arg)))
+        click.echo(f"arg:{'|'.join(arg)}")
 
     result = runner.invoke(cmd2, [])
     assert result.exit_code == 2
@@ -215,7 +209,7 @@ def test_missing_arg(runner):
     @click.command()
     @click.argument("arg")
     def cmd(arg):
-        click.echo("arg:{}".format(arg))
+        click.echo(f"arg:{arg}")
 
     result = runner.invoke(cmd, [])
     assert result.exit_code == 2
@@ -268,7 +262,7 @@ def test_nargs_star_ordering(runner):
             click.echo(arg)
 
     result = runner.invoke(cmd, ["a", "b", "c"])
-    assert result.output.splitlines() == ["(u'a',)" if PY2 else "('a',)", "b", "c"]
+    assert result.output.splitlines() == ["('a',)", "b", "c"]
 
 
 def test_nargs_specified_plus_star_ordering(runner):
@@ -281,11 +275,7 @@ def test_nargs_specified_plus_star_ordering(runner):
             click.echo(arg)
 
     result = runner.invoke(cmd, ["a", "b", "c", "d", "e", "f"])
-    assert result.output.splitlines() == [
-        "(u'a', u'b', u'c')" if PY2 else "('a', 'b', 'c')",
-        "d",
-        "(u'e', u'f')" if PY2 else "('e', 'f')",
-    ]
+    assert result.output.splitlines() == ["('a', 'b', 'c')", "d", "('e', 'f')"]
 
 
 def test_defaults_for_nargs(runner):
diff --git a/tests/test_bashcomplete.py b/tests/test_bashcomplete.py
index df4ef287cba73256bcc2c167a7939ea27cd158df..f48927a5445e16ceebeb0536b32dd41e7ecf0022 100644
--- a/tests/test_bashcomplete.py
+++ b/tests/test_bashcomplete.py
@@ -1,4 +1,3 @@
-# -*- coding: utf-8 -*-
 import pytest
 
 import click
diff --git a/tests/test_basic.py b/tests/test_basic.py
index f07b6d1f50065dc89c0ee2a50e414b4da996ca79..c33e3a4ac898636d547093f957181db73d14f609 100644
--- a/tests/test_basic.py
+++ b/tests/test_basic.py
@@ -1,4 +1,3 @@
-# -*- coding: utf-8 -*-
 import os
 import uuid
 
@@ -82,7 +81,7 @@ def test_basic_option(runner):
     @click.command()
     @click.option("--foo", default="no value")
     def cli(foo):
-        click.echo(u"FOO:[{}]".format(foo))
+        click.echo(f"FOO:[{foo}]")
 
     result = runner.invoke(cli, [])
     assert not result.exception
@@ -100,16 +99,16 @@ def test_basic_option(runner):
     assert not result.exception
     assert "FOO:[]" in result.output
 
-    result = runner.invoke(cli, [u"--foo=\N{SNOWMAN}"])
+    result = runner.invoke(cli, ["--foo=\N{SNOWMAN}"])
     assert not result.exception
-    assert u"FOO:[\N{SNOWMAN}]" in result.output
+    assert "FOO:[\N{SNOWMAN}]" in result.output
 
 
 def test_int_option(runner):
     @click.command()
     @click.option("--foo", default=42)
     def cli(foo):
-        click.echo("FOO:[{}]".format(foo * 2))
+        click.echo(f"FOO:[{foo * 2}]")
 
     result = runner.invoke(cli, [])
     assert not result.exception
@@ -131,7 +130,7 @@ def test_uuid_option(runner):
     )
     def cli(u):
         assert type(u) is uuid.UUID
-        click.echo("U:[{}]".format(u))
+        click.echo(f"U:[{u}]")
 
     result = runner.invoke(cli, [])
     assert not result.exception
@@ -151,7 +150,7 @@ def test_float_option(runner):
     @click.option("--foo", default=42, type=click.FLOAT)
     def cli(foo):
         assert type(foo) is float
-        click.echo("FOO:[{}]".format(foo))
+        click.echo(f"FOO:[{foo}]")
 
     result = runner.invoke(cli, [])
     assert not result.exception
@@ -182,7 +181,7 @@ def test_boolean_option(runner):
         assert result.output == "False\n"
         result = runner.invoke(cli, [])
         assert not result.exception
-        assert result.output == "{}\n".format(default)
+        assert result.output == f"{default}\n"
 
     for default in True, False:
 
@@ -193,10 +192,10 @@ def test_boolean_option(runner):
 
         result = runner.invoke(cli, ["--flag"])
         assert not result.exception
-        assert result.output == "{}\n".format(not default)
+        assert result.output == f"{not default}\n"
         result = runner.invoke(cli, [])
         assert not result.exception
-        assert result.output == "{}\n".format(default)
+        assert result.output == f"{default}\n"
 
 
 def test_boolean_conversion(runner):
@@ -219,7 +218,7 @@ def test_boolean_conversion(runner):
 
         result = runner.invoke(cli, [])
         assert not result.exception
-        assert result.output == "{}\n".format(default)
+        assert result.output == f"{default}\n"
 
 
 def test_file_option(runner):
@@ -308,8 +307,8 @@ def test_path_option(runner):
     @click.command()
     @click.option("-f", type=click.Path(exists=True))
     def showtype(f):
-        click.echo("is_file={}".format(os.path.isfile(f)))
-        click.echo("is_dir={}".format(os.path.isdir(f)))
+        click.echo(f"is_file={os.path.isfile(f)}")
+        click.echo(f"is_dir={os.path.isdir(f)}")
 
     with runner.isolated_filesystem():
         result = runner.invoke(showtype, ["-f", "xxx"])
@@ -322,7 +321,7 @@ def test_path_option(runner):
     @click.command()
     @click.option("-f", type=click.Path())
     def exists(f):
-        click.echo("exists={}".format(os.path.exists(f)))
+        click.echo(f"exists={os.path.exists(f)}")
 
     with runner.isolated_filesystem():
         result = runner.invoke(exists, ["-f", "xxx"])
diff --git a/tests/test_chain.py b/tests/test_chain.py
index c227270f22b55b48dcfe44c3b763aca9c65eb5f6..0462779091fcccb8248135b3276767187c93e181 100644
--- a/tests/test_chain.py
+++ b/tests/test_chain.py
@@ -7,9 +7,8 @@ import click
 
 def debug():
     click.echo(
-        "{}={}".format(
-            sys._getframe(1).f_code.co_name, "|".join(click.get_current_context().args)
-        )
+        f"{sys._getframe(1).f_code.co_name}"
+        f"={'|'.join(click.get_current_context().args)}"
     )
 
 
@@ -77,12 +76,12 @@ def test_chaining_with_options(runner):
     @cli.command("sdist")
     @click.option("--format")
     def sdist(format):
-        click.echo("sdist called {}".format(format))
+        click.echo(f"sdist called {format}")
 
     @cli.command("bdist")
     @click.option("--format")
     def bdist(format):
-        click.echo("bdist called {}".format(format))
+        click.echo(f"bdist called {format}")
 
     result = runner.invoke(cli, ["bdist", "--format=1", "sdist", "--format=2"])
     assert not result.exception
@@ -97,12 +96,12 @@ def test_chaining_with_arguments(runner):
     @cli.command("sdist")
     @click.argument("format")
     def sdist(format):
-        click.echo("sdist called {}".format(format))
+        click.echo(f"sdist called {format}")
 
     @cli.command("bdist")
     @click.argument("format")
     def bdist(format):
-        click.echo("bdist called {}".format(format))
+        click.echo(f"bdist called {format}")
 
     result = runner.invoke(cli, ["bdist", "1", "sdist", "2"])
     assert not result.exception
@@ -192,7 +191,7 @@ def test_multicommand_arg_behavior(runner):
     @click.group(chain=True)
     @click.argument("arg")
     def cli(arg):
-        click.echo("cli:{}".format(arg))
+        click.echo(f"cli:{arg}")
 
     @cli.command()
     def a():
diff --git a/tests/test_commands.py b/tests/test_commands.py
index 1d99218cdc6b2287a8633fe809d4b7e41f65434d..eace114a4c21ec5e3a84a8adf7b27407a6b6598d 100644
--- a/tests/test_commands.py
+++ b/tests/test_commands.py
@@ -1,4 +1,3 @@
-# -*- coding: utf-8 -*-
 import re
 
 import click
@@ -26,7 +25,7 @@ def test_other_command_forward(runner):
     @cli.command()
     @click.option("--count", default=1)
     def test(count):
-        click.echo("Count: {:d}".format(count))
+        click.echo(f"Count: {count:d}")
 
     @cli.command()
     @click.option("--count", default=1)
@@ -102,7 +101,7 @@ def test_group_with_args(runner):
     @click.group()
     @click.argument("obj")
     def cli(obj):
-        click.echo("obj={}".format(obj))
+        click.echo(f"obj={obj}")
 
     @cli.command()
     def move():
@@ -211,7 +210,7 @@ def test_object_propagation(runner):
         @cli.command()
         @click.pass_context
         def sync(ctx):
-            click.echo("Debug is {}".format("on" if ctx.obj["DEBUG"] else "off"))
+            click.echo(f"Debug is {'on' if ctx.obj['DEBUG'] else 'off'}")
 
         result = runner.invoke(cli, ["sync"])
         assert result.exception is None
@@ -264,8 +263,8 @@ def test_unprocessed_options(runner):
     @click.argument("args", nargs=-1, type=click.UNPROCESSED)
     @click.option("--verbose", "-v", count=True)
     def cli(verbose, args):
-        click.echo("Verbosity: {}".format(verbose))
-        click.echo("Args: {}".format("|".join(args)))
+        click.echo(f"Verbosity: {verbose}")
+        click.echo(f"Args: {'|'.join(args)}")
 
     result = runner.invoke(cli, ["-foo", "-vvvvx", "--muhaha", "x", "y", "-x"])
     assert not result.exception
diff --git a/tests/test_compat.py b/tests/test_compat.py
index 7851ab8ecd89bb7670d3cb9db6610c2262611ea4..3c458e9c7240daaef813e85e58983569e4f3fa96 100644
--- a/tests/test_compat.py
+++ b/tests/test_compat.py
@@ -1,25 +1,9 @@
 import pytest
 
-import click
 from click._compat import should_strip_ansi
 from click._compat import WIN
 
 
-def test_legacy_callbacks(runner):
-    def legacy_callback(ctx, value):
-        return value.upper()
-
-    @click.command()
-    @click.option("--foo", callback=legacy_callback)
-    def cli(foo):
-        click.echo(foo)
-
-    with pytest.warns(DeprecationWarning, match="2-arg style"):
-        result = runner.invoke(cli, ["--foo", "wat"])
-        assert result.exit_code == 0
-        assert "WAT" in result.output
-
-
 def test_bash_func_name():
     from click._bashcomplete import get_completion_script
 
@@ -39,7 +23,7 @@ def test_zsh_func_name():
 
 @pytest.mark.xfail(WIN, reason="Jupyter not tested/supported on Windows")
 def test_is_jupyter_kernel_output():
-    class JupyterKernelFakeStream(object):
+    class JupyterKernelFakeStream:
         pass
 
     # implementation detail, aka cheapskate test
diff --git a/tests/test_context.py b/tests/test_context.py
index b3a20c6a5a5626ee648166f2a8d1d680811a58b6..44befa5843987fa48c51f6f3ca3d2cf8f57f2705 100644
--- a/tests/test_context.py
+++ b/tests/test_context.py
@@ -1,9 +1,11 @@
-# -*- coding: utf-8 -*-
+import pytest
+
 import click
+from click.core import ParameterSource
 
 
 def test_ensure_context_objects(runner):
-    class Foo(object):
+    class Foo:
         def __init__(self):
             self.title = "default"
 
@@ -25,7 +27,7 @@ def test_ensure_context_objects(runner):
 
 
 def test_get_context_objects(runner):
-    class Foo(object):
+    class Foo:
         def __init__(self):
             self.title = "default"
 
@@ -48,7 +50,7 @@ def test_get_context_objects(runner):
 
 
 def test_get_context_objects_no_ensuring(runner):
-    class Foo(object):
+    class Foo:
         def __init__(self):
             self.title = "default"
 
@@ -71,7 +73,7 @@ def test_get_context_objects_no_ensuring(runner):
 
 
 def test_get_context_objects_missing(runner):
-    class Foo(object):
+    class Foo:
         pass
 
     pass_foo = click.make_pass_decorator(Foo)
@@ -129,7 +131,7 @@ def test_global_context_object(runner):
 
 
 def test_context_meta(runner):
-    LANG_KEY = "{}.lang".format(__name__)
+    LANG_KEY = f"{__name__}.lang"
 
     def set_language(value):
         click.get_current_context().meta[LANG_KEY] = value
@@ -216,7 +218,7 @@ def test_make_pass_decorator_args(runner):
     invocation order.
     """
 
-    class Foo(object):
+    class Foo:
         title = "foocmd"
 
     pass_foo = click.make_pass_decorator(Foo)
@@ -261,3 +263,65 @@ def test_exit_not_standalone():
         ctx.exit(0)
 
     assert cli.main([], "test_exit_not_standalone", standalone_mode=False) == 0
+
+
+def test_parameter_source_default(runner):
+    @click.command()
+    @click.pass_context
+    @click.option("-o", "--option", default=1)
+    def cli(ctx, option):
+        click.echo(ctx.get_parameter_source("option"))
+
+    rv = runner.invoke(cli)
+    assert rv.output.rstrip() == ParameterSource.DEFAULT
+
+
+def test_parameter_source_default_map(runner):
+    @click.command()
+    @click.pass_context
+    @click.option("-o", "--option", default=1)
+    def cli(ctx, option):
+        click.echo(ctx.get_parameter_source("option"))
+
+    rv = runner.invoke(cli, default_map={"option": 1})
+    assert rv.output.rstrip() == ParameterSource.DEFAULT_MAP
+
+
+def test_parameter_source_commandline(runner):
+    @click.command()
+    @click.pass_context
+    @click.option("-o", "--option", default=1)
+    def cli(ctx, option):
+        click.echo(ctx.get_parameter_source("option"))
+
+    rv = runner.invoke(cli, ["-o", "1"])
+    assert rv.output.rstrip() == ParameterSource.COMMANDLINE
+    rv = runner.invoke(cli, ["--option", "1"])
+    assert rv.output.rstrip() == ParameterSource.COMMANDLINE
+
+
+def test_parameter_source_environment(runner):
+    @click.command()
+    @click.pass_context
+    @click.option("-o", "--option", default=1)
+    def cli(ctx, option):
+        click.echo(ctx.get_parameter_source("option"))
+
+    rv = runner.invoke(cli, auto_envvar_prefix="TEST", env={"TEST_OPTION": "1"})
+    assert rv.output.rstrip() == ParameterSource.ENVIRONMENT
+
+
+def test_parameter_source_environment_variable_specified(runner):
+    @click.command()
+    @click.pass_context
+    @click.option("-o", "--option", default=1, envvar="NAME")
+    def cli(ctx, option):
+        click.echo(ctx.get_parameter_source("option"))
+
+    rv = runner.invoke(cli, env={"NAME": "1"})
+    assert rv.output.rstrip() == ParameterSource.ENVIRONMENT
+
+
+def test_validate_parameter_source():
+    with pytest.raises(ValueError):
+        ParameterSource.validate("NOT_A_VALID_PARAMETER_SOURCE")
diff --git a/tests/test_defaults.py b/tests/test_defaults.py
index ce449033199a68f6724f560fceeefe12be724d85..0e438eb89122cc49b5c1c34490ff818d47fc34f5 100644
--- a/tests/test_defaults.py
+++ b/tests/test_defaults.py
@@ -6,7 +6,7 @@ def test_basic_defaults(runner):
     @click.option("--foo", default=42, type=click.FLOAT)
     def cli(foo):
         assert type(foo) is float
-        click.echo("FOO:[{}]".format(foo))
+        click.echo(f"FOO:[{foo}]")
 
     result = runner.invoke(cli, [])
     assert not result.exception
@@ -32,8 +32,8 @@ def test_nargs_plus_multiple(runner):
         "--arg", default=((1, 2), (3, 4)), nargs=2, multiple=True, type=click.INT
     )
     def cli(arg):
-        for item in arg:
-            click.echo("<{0[0]:d}|{0[1]:d}>".format(item))
+        for a, b in arg:
+            click.echo(f"<{a:d}|{b:d}>")
 
     result = runner.invoke(cli, [])
     assert not result.exception
diff --git a/tests/test_formatting.py b/tests/test_formatting.py
index 4fabbb20f20b24d456ba11a8ad247d9d43920d4f..b2f72b7d248bcd4b6e9c70c95d3d1742fe3dfa6a 100644
--- a/tests/test_formatting.py
+++ b/tests/test_formatting.py
@@ -1,4 +1,3 @@
-# -*- coding: utf-8 -*-
 import click
 
 
@@ -151,7 +150,7 @@ def test_formatting_usage_error(runner):
     @click.command()
     @click.argument("arg")
     def cmd(arg):
-        click.echo("arg:{}".format(arg))
+        click.echo(f"arg:{arg}")
 
     result = runner.invoke(cmd, [])
     assert result.exit_code == 2
@@ -208,7 +207,7 @@ def test_formatting_usage_error_nested(runner):
     @cmd.command()
     @click.argument("bar")
     def foo(bar):
-        click.echo("foo:{}".format(bar))
+        click.echo(f"foo:{bar}")
 
     result = runner.invoke(cmd, ["foo"])
     assert result.exit_code == 2
@@ -224,7 +223,7 @@ def test_formatting_usage_error_no_help(runner):
     @click.command(add_help_option=False)
     @click.argument("arg")
     def cmd(arg):
-        click.echo("arg:{}".format(arg))
+        click.echo(f"arg:{arg}")
 
     result = runner.invoke(cmd, [])
     assert result.exit_code == 2
@@ -239,7 +238,7 @@ def test_formatting_usage_custom_help(runner):
     @click.command(context_settings=dict(help_option_names=["--man"]))
     @click.argument("arg")
     def cmd(arg):
-        click.echo("arg:{}".format(arg))
+        click.echo(f"arg:{arg}")
 
     result = runner.invoke(cmd, [])
     assert result.exit_code == 2
diff --git a/tests/test_imports.py b/tests/test_imports.py
index b99d453d0626e9550b85e7703330d1ca0cd9d987..be8730f64e4b5ac263ac7912a18d50dcc758f4c9 100644
--- a/tests/test_imports.py
+++ b/tests/test_imports.py
@@ -6,10 +6,7 @@ from click._compat import WIN
 
 
 IMPORT_TEST = b"""\
-try:
-    import __builtin__ as builtins
-except ImportError:
-    import builtins
+import builtins
 
 found_imports = set()
 real_import = builtins.__import__
@@ -61,9 +58,7 @@ def test_light_imports():
         [sys.executable, "-"], stdin=subprocess.PIPE, stdout=subprocess.PIPE
     )
     rv = c.communicate(IMPORT_TEST)[0]
-
-    if sys.version_info[0] != 2:
-        rv = rv.decode("utf-8")
+    rv = rv.decode("utf-8")
     imported = json.loads(rv)
 
     for module in imported:
diff --git a/tests/test_options.py b/tests/test_options.py
index 4baa37451ff3c94884a2d56b63166a0d91939f76..1b9b27e0e075dcfdd785411d95d347c66ff1cb1b 100644
--- a/tests/test_options.py
+++ b/tests/test_options.py
@@ -1,11 +1,9 @@
-# -*- coding: utf-8 -*-
 import os
 import re
 
 import pytest
 
 import click
-from click._compat import text_type
 
 
 def test_prefixes(runner):
@@ -13,7 +11,7 @@ def test_prefixes(runner):
     @click.option("++foo", is_flag=True, help="das foo")
     @click.option("--bar", is_flag=True, help="das bar")
     def cli(foo, bar):
-        click.echo("foo={} bar={}".format(foo, bar))
+        click.echo(f"foo={foo} bar={bar}")
 
     result = runner.invoke(cli, ["++foo", "--bar"])
     assert not result.exception
@@ -46,8 +44,8 @@ def test_nargs_tup_composite_mult(runner):
     @click.command()
     @click.option("--item", type=(str, int), multiple=True)
     def copy(item):
-        for item in item:
-            click.echo("name={0[0]} id={0[1]:d}".format(item))
+        for name, id in item:
+            click.echo(f"name={name} id={id:d}")
 
     result = runner.invoke(copy, ["--item", "peter", "1", "--item", "max", "2"])
     assert not result.exception
@@ -58,7 +56,7 @@ def test_counting(runner):
     @click.command()
     @click.option("-v", count=True, help="Verbosity", type=click.IntRange(0, 3))
     def cli(v):
-        click.echo("verbosity={:d}".format(v))
+        click.echo(f"verbosity={v:d}")
 
     result = runner.invoke(cli, ["-vvv"])
     assert not result.exception
@@ -87,7 +85,7 @@ def test_unknown_options(runner, unknown_flag):
 
     result = runner.invoke(cli, [unknown_flag])
     assert result.exception
-    assert "no such option: {}".format(unknown_flag) in result.output
+    assert f"no such option: {unknown_flag}" in result.output
 
 
 def test_multiple_required(runner):
@@ -109,7 +107,7 @@ def test_empty_envvar(runner):
     @click.command()
     @click.option("--mypath", type=click.Path(exists=True), envvar="MYPATH")
     def cli(mypath):
-        click.echo("mypath: {}".format(mypath))
+        click.echo(f"mypath: {mypath}")
 
     result = runner.invoke(cli, [], env={"MYPATH": ""})
     assert result.exit_code == 0
@@ -146,7 +144,7 @@ def test_multiple_envvar(runner):
         cmd,
         [],
         auto_envvar_prefix="TEST",
-        env={"TEST_ARG": "foo{}bar".format(os.path.pathsep)},
+        env={"TEST_ARG": f"foo{os.path.pathsep}bar"},
     )
     assert not result.exception
     assert result.output == "foo|bar\n"
@@ -170,11 +168,11 @@ def test_multiple_default_type(runner):
     @click.option("--arg1", multiple=True, default=("foo", "bar"))
     @click.option("--arg2", multiple=True, default=(1, "a"))
     def cmd(arg1, arg2):
-        assert all(isinstance(e[0], text_type) for e in arg1)
-        assert all(isinstance(e[1], text_type) for e in arg1)
+        assert all(isinstance(e[0], str) for e in arg1)
+        assert all(isinstance(e[1], str) for e in arg1)
 
         assert all(isinstance(e[0], int) for e in arg2)
-        assert all(isinstance(e[1], text_type) for e in arg2)
+        assert all(isinstance(e[1], str) for e in arg2)
 
     result = runner.invoke(
         cmd, "--arg1 a b --arg1 test 1 --arg2 2 two --arg2 4 four".split()
@@ -423,12 +421,13 @@ def test_option_help_preserve_paragraphs(runner):
 
     result = runner.invoke(cmd, ["--help"],)
     assert result.exit_code == 0
+    i = " " * 21
     assert (
         "  -C, --config PATH  Configuration file to use.\n"
-        "{i}\n"
-        "{i}If not given, the environment variable CONFIG_FILE is\n"
-        "{i}consulted and used if set. If neither are given, a default\n"
-        "{i}configuration file is loaded.".format(i=" " * 21)
+        f"{i}\n"
+        f"{i}If not given, the environment variable CONFIG_FILE is\n"
+        f"{i}consulted and used if set. If neither are given, a default\n"
+        f"{i}configuration file is loaded."
     ) in result.output
 
 
diff --git a/tests/test_termui.py b/tests/test_termui.py
index 99a6a476fb17287524c339c770f8eec611b4db87..7165023b46df9a96b22f809adb77b039f28260f1 100644
--- a/tests/test_termui.py
+++ b/tests/test_termui.py
@@ -1,4 +1,3 @@
-# -*- coding: utf-8 -*-
 import time
 
 import pytest
@@ -7,7 +6,7 @@ import click._termui_impl
 from click._compat import WIN
 
 
-class FakeClock(object):
+class FakeClock:
     def __init__(self):
         self.now = time.time()
 
@@ -42,7 +41,7 @@ def test_progressbar_strip_regression(runner, monkeypatch):
 
 
 def test_progressbar_length_hint(runner, monkeypatch):
-    class Hinted(object):
+    class Hinted:
         def __init__(self, n):
             self.items = list(range(n))
 
@@ -123,7 +122,7 @@ def test_progressbar_format_pos(runner, pos, length):
     with _create_progress(length, length_known=length != 0, pos=pos) as progress:
         result = progress.format_pos()
         if progress.length_known:
-            assert result == "{}/{}".format(pos, length)
+            assert result == f"{pos}/{length}"
         else:
             assert result == str(pos)
 
@@ -248,7 +247,7 @@ def test_file_prompt_default_format(runner, file_kwargs):
         click.echo(f.name)
 
     result = runner.invoke(cli)
-    assert result.output == "file [{0}]: \n{0}\n".format(__file__)
+    assert result.output == f"file [{__file__}]: \n{__file__}\n"
 
 
 def test_secho(runner):
@@ -285,7 +284,55 @@ def test_progressbar_update(runner, monkeypatch):
     assert "100%          " in lines[3]
 
 
-@pytest.mark.parametrize("key_char", (u"h", u"H", u"é", u"À", u" ", u"字", u"àH", u"àR"))
+def test_progressbar_item_show_func(runner, monkeypatch):
+    fake_clock = FakeClock()
+
+    @click.command()
+    def cli():
+        with click.progressbar(
+            range(4), item_show_func=lambda x: f"Custom {x}"
+        ) as progress:
+            for _ in progress:
+                fake_clock.advance_time()
+                print("")
+
+    monkeypatch.setattr(time, "time", fake_clock.time)
+    monkeypatch.setattr(click._termui_impl, "isatty", lambda _: True)
+    output = runner.invoke(cli, []).output
+
+    lines = [line for line in output.split("\n") if "[" in line]
+
+    assert "Custom 0" in lines[0]
+    assert "Custom 1" in lines[1]
+    assert "Custom 2" in lines[2]
+    assert "Custom None" in lines[3]
+
+
+def test_progressbar_update_with_item_show_func(runner, monkeypatch):
+    fake_clock = FakeClock()
+
+    @click.command()
+    def cli():
+        with click.progressbar(
+            length=6, item_show_func=lambda x: f"Custom {x}"
+        ) as progress:
+            while not progress.finished:
+                fake_clock.advance_time()
+                progress.update(2, progress.pos)
+                print("")
+
+    monkeypatch.setattr(time, "time", fake_clock.time)
+    monkeypatch.setattr(click._termui_impl, "isatty", lambda _: True)
+    output = runner.invoke(cli, []).output
+
+    lines = [line for line in output.split("\n") if "[" in line]
+
+    assert "Custom 0" in lines[0]
+    assert "Custom 2" in lines[1]
+    assert "Custom 4" in lines[2]
+
+
+@pytest.mark.parametrize("key_char", ("h", "H", "é", "À", " ", "字", "àH", "àR"))
 @pytest.mark.parametrize("echo", [True, False])
 @pytest.mark.skipif(not WIN, reason="Tests user-input using the msvcrt module.")
 def test_getchar_windows(runner, monkeypatch, key_char, echo):
@@ -296,7 +343,7 @@ def test_getchar_windows(runner, monkeypatch, key_char, echo):
 
 
 @pytest.mark.parametrize(
-    "special_key_char, key_char", [(u"\x00", "a"), (u"\x00", "b"), (u"\xe0", "c")]
+    "special_key_char, key_char", [("\x00", "a"), ("\x00", "b"), ("\xe0", "c")]
 )
 @pytest.mark.skipif(
     not WIN, reason="Tests special character inputs using the msvcrt module."
@@ -307,11 +354,11 @@ def test_getchar_special_key_windows(runner, monkeypatch, special_key_char, key_
         click._termui_impl.msvcrt, "getwch", lambda: ordered_inputs.pop()
     )
     monkeypatch.setattr(click.termui, "_getchar", None)
-    assert click.getchar() == special_key_char + key_char
+    assert click.getchar() == f"{special_key_char}{key_char}"
 
 
 @pytest.mark.parametrize(
-    ("key_char", "exc"), [(u"\x03", KeyboardInterrupt), (u"\x1a", EOFError)],
+    ("key_char", "exc"), [("\x03", KeyboardInterrupt), ("\x1a", EOFError)],
 )
 @pytest.mark.skipif(not WIN, reason="Tests user-input using the msvcrt module.")
 def test_getchar_windows_exceptions(runner, monkeypatch, key_char, exc):
diff --git a/tests/test_testing.py b/tests/test_testing.py
index 22a285d906b3bec64441427ca7e0f6f566eafac9..7360473f3bfcaf2dd358ab61b7572706230d6b0b 100644
--- a/tests/test_testing.py
+++ b/tests/test_testing.py
@@ -1,19 +1,13 @@
 import os
 import sys
+from io import BytesIO
 
 import pytest
 
 import click
-from click._compat import PY2
 from click._compat import WIN
 from click.testing import CliRunner
 
-# Use the most reasonable io that users would use for the python version.
-if PY2:
-    from cStringIO import StringIO as ReasonableBytesIO
-else:
-    from io import BytesIO as ReasonableBytesIO
-
 
 def test_runner():
     @click.command()
@@ -51,12 +45,12 @@ def test_runner_with_stream():
             o.flush()
 
     runner = CliRunner()
-    result = runner.invoke(test, input=ReasonableBytesIO(b"Hello World!\n"))
+    result = runner.invoke(test, input=BytesIO(b"Hello World!\n"))
     assert not result.exception
     assert result.output == "Hello World!\n"
 
     runner = CliRunner(echo_stdin=True)
-    result = runner.invoke(test, input=ReasonableBytesIO(b"Hello World!\n"))
+    result = runner.invoke(test, input=BytesIO(b"Hello World!\n"))
     assert not result.exception
     assert result.output == "Hello World!\nHello World!\n"
 
@@ -65,7 +59,7 @@ def test_prompts():
     @click.command()
     @click.option("--foo", prompt=True)
     def test(foo):
-        click.echo("foo={}".format(foo))
+        click.echo(f"foo={foo}")
 
     runner = CliRunner()
     result = runner.invoke(test, input="wau wau\n")
@@ -75,7 +69,7 @@ def test_prompts():
     @click.command()
     @click.option("--foo", prompt=True, hide_input=True)
     def test(foo):
-        click.echo("foo={}".format(foo))
+        click.echo(f"foo={foo}")
 
     runner = CliRunner()
     result = runner.invoke(test, input="wau wau\n")
@@ -131,7 +125,7 @@ def test_with_color():
     assert not result.exception
 
     result = runner.invoke(cli, color=True)
-    assert result.output == "{}\n".format(click.style("hello world", fg="blue"))
+    assert result.output == f"{click.style('hello world', fg='blue')}\n"
     assert not result.exception
 
 
@@ -219,7 +213,7 @@ def test_exit_code_and_output_from_sys_exit():
 def test_env():
     @click.command()
     def cli_env():
-        click.echo("ENV={}".format(os.environ["TEST_CLICK_ENV"]))
+        click.echo(f"ENV={os.environ['TEST_CLICK_ENV']}")
 
     runner = CliRunner()
 
diff --git a/tests/test_utils.py b/tests/test_utils.py
index d86f59b8179225b33457a4a4f12f5a74c8c79b43..bf3956d1f6821df9ef270ec01ef9ecfbad424050 100644
--- a/tests/test_utils.py
+++ b/tests/test_utils.py
@@ -1,6 +1,7 @@
 import os
 import stat
 import sys
+from io import StringIO
 
 import pytest
 
@@ -11,7 +12,7 @@ from click._compat import WIN
 
 def test_echo(runner):
     with runner.isolation() as outstreams:
-        click.echo(u"\N{SNOWMAN}")
+        click.echo("\N{SNOWMAN}")
         click.echo(b"\x44\x44")
         click.echo(42, nl=False)
         click.echo(b"a", nl=False)
@@ -19,19 +20,7 @@ def test_echo(runner):
         bytes = outstreams[0].getvalue().replace(b"\r\n", b"\n")
         assert bytes == b"\xe2\x98\x83\nDD\n42ax"
 
-    # If we are in Python 2, we expect that writing bytes into a string io
-    # does not do anything crazy.  In Python 3
-    if sys.version_info[0] == 2:
-        import StringIO
-
-        sys.stdout = x = StringIO.StringIO()
-        try:
-            click.echo("\xf6")
-        finally:
-            sys.stdout = sys.__stdout__
-        assert x.getvalue() == "\xf6\n"
-
-    # And in any case, if wrapped, we expect bytes to survive.
+    # if wrapped, we expect bytes to survive.
     @click.command()
     def cli():
         click.echo(b"\xf6")
@@ -49,8 +38,8 @@ def test_echo_custom_file():
     import io
 
     f = io.StringIO()
-    click.echo(u"hello", file=f)
-    assert f.getvalue() == u"hello\n"
+    click.echo("hello", file=f)
+    assert f.getvalue() == "hello\n"
 
 
 @pytest.mark.parametrize(
@@ -91,14 +80,12 @@ def test_unstyle_other_ansi(text, expect):
 def test_filename_formatting():
     assert click.format_filename(b"foo.txt") == "foo.txt"
     assert click.format_filename(b"/x/foo.txt") == "/x/foo.txt"
-    assert click.format_filename(u"/x/foo.txt") == "/x/foo.txt"
-    assert click.format_filename(u"/x/foo.txt", shorten=True) == "foo.txt"
+    assert click.format_filename("/x/foo.txt") == "/x/foo.txt"
+    assert click.format_filename("/x/foo.txt", shorten=True) == "foo.txt"
 
     # filesystem encoding on windows permits this.
     if not WIN:
-        assert (
-            click.format_filename(b"/x/foo\xff.txt", shorten=True) == u"foo\ufffd.txt"
-        )
+        assert click.format_filename(b"/x/foo\xff.txt", shorten=True) == "foo\ufffd.txt"
 
 
 def test_prompts(runner):
@@ -203,31 +190,27 @@ def test_echo_color_flag(monkeypatch, capfd):
 
     click.echo(styled_text, color=False)
     out, err = capfd.readouterr()
-    assert out == "{}\n".format(text)
+    assert out == f"{text}\n"
 
     click.echo(styled_text, color=True)
     out, err = capfd.readouterr()
-    assert out == "{}\n".format(styled_text)
+    assert out == f"{styled_text}\n"
 
     isatty = True
     click.echo(styled_text)
     out, err = capfd.readouterr()
-    assert out == "{}\n".format(styled_text)
+    assert out == f"{styled_text}\n"
 
     isatty = False
     click.echo(styled_text)
     out, err = capfd.readouterr()
-    assert out == "{}\n".format(text)
+    assert out == f"{text}\n"
 
 
 @pytest.mark.skipif(WIN, reason="Test too complex to make work windows.")
 def test_echo_writing_to_standard_error(capfd, monkeypatch):
     def emulate_input(text):
         """Emulate keyboard input."""
-        if sys.version_info[0] == 2:
-            from StringIO import StringIO
-        else:
-            from io import StringIO
         monkeypatch.setattr(sys, "stdin", StringIO(text))
 
     click.echo("Echo to standard output")
