diff --git a/src/test/java/org/apache/commons/lang3/StringUtilsTest.java b/src/test/java/org/apache/commons/lang3/StringUtilsTest.java
index f11ec703daadedc63ad32f74ff8edd668be02ff5..44200bd92540bc4ace6b48666b44d8981920cabe 100644
--- a/src/test/java/org/apache/commons/lang3/StringUtilsTest.java
+++ b/src/test/java/org/apache/commons/lang3/StringUtilsTest.java
@@ -460,6 +460,24 @@ public class StringUtilsTest {
         assertEquals(TEXT_LIST, StringUtils.join(Arrays.asList(ARRAY_LIST), SEPARATOR));
     }
 
+    @Test
+    public void testJoinWith() {
+        assertEquals("", StringUtils.joinWith(",", new Object[0]));        // empty array
+        assertEquals("", StringUtils.joinWith(",", NULL_ARRAY_LIST));
+        assertEquals("null", StringUtils.joinWith(",", NULL_TO_STRING_LIST));   //toString method prints 'null'
+
+        assertEquals("a,b,c", StringUtils.joinWith(",", new String[]{"a", "b", "c"}));
+        assertEquals(",a,", StringUtils.joinWith(",", new String[]{null, "a", ""}));
+
+        assertEquals("ab", StringUtils.joinWith(null, "a", "b"));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testJoinWithThrowsException() {
+        StringUtils.joinWith(",", null);
+    }
+
+
     @Test
     public void testSplit_String() {
         assertNull(StringUtils.split(null));
