diff --git a/src/test/java/org/apache/commons/lang3/StringUtilsTest.java b/src/test/java/org/apache/commons/lang3/StringUtilsTest.java
index 1fa164ea45d0061cbb94dac45348ff26336f95cf..df48c272256058241d4cb7cd5ab0cdf5b4580c66 100644
--- a/src/test/java/org/apache/commons/lang3/StringUtilsTest.java
+++ b/src/test/java/org/apache/commons/lang3/StringUtilsTest.java
@@ -1208,6 +1208,33 @@ public class StringUtilsTest {
         assertEquals("farfarfar", StringUtils.replace("foofoofoo", "oo", "ar"));
     }
 
+    @Test
+    public void testReplaceIgnoreCase_StringStringString() {
+        assertEquals(null, StringUtils.replaceIgnoreCase(null, null, null));
+        assertEquals(null, StringUtils.replaceIgnoreCase(null, null, "any"));
+        assertEquals(null, StringUtils.replaceIgnoreCase(null, "any", null));
+        assertEquals(null, StringUtils.replaceIgnoreCase(null, "any", "any"));
+
+        assertEquals("", StringUtils.replaceIgnoreCase("", null, null));
+        assertEquals("", StringUtils.replaceIgnoreCase("", null, "any"));
+        assertEquals("", StringUtils.replaceIgnoreCase("", "any", null));
+        assertEquals("", StringUtils.replaceIgnoreCase("", "any", "any"));
+
+        assertEquals("FOO", StringUtils.replaceIgnoreCase("FOO", "", "any"));
+        assertEquals("FOO", StringUtils.replaceIgnoreCase("FOO", null, "any"));
+        assertEquals("FOO", StringUtils.replaceIgnoreCase("FOO", "F", null));
+        assertEquals("FOO", StringUtils.replaceIgnoreCase("FOO", null, null));
+
+        assertEquals("", StringUtils.replaceIgnoreCase("foofoofoo", "foo", ""));
+        assertEquals("barbarbar", StringUtils.replaceIgnoreCase("foofoofoo", "foo", "bar"));
+        assertEquals("farfarfar", StringUtils.replaceIgnoreCase("foofoofoo", "oo", "ar"));
+
+        // IgnoreCase
+        assertEquals("", StringUtils.replaceIgnoreCase("foofoofoo", "FOO", ""));
+        assertEquals("barbarbar", StringUtils.replaceIgnoreCase("fooFOOfoo", "foo", "bar"));
+        assertEquals("farfarfar", StringUtils.replaceIgnoreCase("foofOOfoo", "OO", "ar"));
+    }
+
     @Test
     public void testReplacePattern() {
         assertNull(StringUtils.replacePattern(null, "", ""));
@@ -1330,6 +1357,43 @@ public class StringUtilsTest {
         assertEquals("f", StringUtils.replace("oofoo", "o", "", 1000));
     }
 
+    @Test
+    public void testReplaceIgnoreCase_StringStringStringInt() {
+        assertEquals(null, StringUtils.replaceIgnoreCase(null, null, null, 2));
+        assertEquals(null, StringUtils.replaceIgnoreCase(null, null, "any", 2));
+        assertEquals(null, StringUtils.replaceIgnoreCase(null, "any", null, 2));
+        assertEquals(null, StringUtils.replaceIgnoreCase(null, "any", "any", 2));
+
+        assertEquals("", StringUtils.replaceIgnoreCase("", null, null, 2));
+        assertEquals("", StringUtils.replaceIgnoreCase("", null, "any", 2));
+        assertEquals("", StringUtils.replaceIgnoreCase("", "any", null, 2));
+        assertEquals("", StringUtils.replaceIgnoreCase("", "any", "any", 2));
+
+        String str = new String(new char[] { 'o', 'o', 'f', 'o', 'o' });
+        assertSame(str, StringUtils.replaceIgnoreCase(str, "x", "", -1));
+
+        assertEquals("f", StringUtils.replaceIgnoreCase("oofoo", "o", "", -1));
+        assertEquals("oofoo", StringUtils.replaceIgnoreCase("oofoo", "o", "", 0));
+        assertEquals("ofoo", StringUtils.replaceIgnoreCase("oofoo", "o", "", 1));
+        assertEquals("foo", StringUtils.replaceIgnoreCase("oofoo", "o", "", 2));
+        assertEquals("fo", StringUtils.replaceIgnoreCase("oofoo", "o", "", 3));
+        assertEquals("f", StringUtils.replaceIgnoreCase("oofoo", "o", "", 4));
+
+        assertEquals("f", StringUtils.replaceIgnoreCase("oofoo", "o", "", -5));
+        assertEquals("f", StringUtils.replaceIgnoreCase("oofoo", "o", "", 1000));
+
+        // IgnoreCase
+        assertEquals("f", StringUtils.replaceIgnoreCase("oofoo", "O", "", -1));
+        assertEquals("oofoo", StringUtils.replaceIgnoreCase("oofoo", "O", "", 0));
+        assertEquals("ofoo", StringUtils.replaceIgnoreCase("oofoo", "O", "", 1));
+        assertEquals("foo", StringUtils.replaceIgnoreCase("oofoo", "O", "", 2));
+        assertEquals("fo", StringUtils.replaceIgnoreCase("oofoo", "O", "", 3));
+        assertEquals("f", StringUtils.replaceIgnoreCase("oofoo", "O", "", 4));
+
+        assertEquals("f", StringUtils.replaceIgnoreCase("oofoo", "O", "", -5));
+        assertEquals("f", StringUtils.replaceIgnoreCase("oofoo", "O", "", 1000));
+    }
+
     @Test
     public void testReplaceOnce_StringStringString() {
         assertNull(StringUtils.replaceOnce(null, null, null));
@@ -1350,6 +1414,29 @@ public class StringUtilsTest {
         assertEquals("foofoo", StringUtils.replaceOnce("foofoofoo", "foo", ""));
     }
 
+    @Test
+    public void testReplaceOnceIgnoreCase_StringStringString() {
+        assertEquals(null, StringUtils.replaceOnceIgnoreCase(null, null, null));
+        assertEquals(null, StringUtils.replaceOnceIgnoreCase(null, null, "any"));
+        assertEquals(null, StringUtils.replaceOnceIgnoreCase(null, "any", null));
+        assertEquals(null, StringUtils.replaceOnceIgnoreCase(null, "any", "any"));
+
+        assertEquals("", StringUtils.replaceOnceIgnoreCase("", null, null));
+        assertEquals("", StringUtils.replaceOnceIgnoreCase("", null, "any"));
+        assertEquals("", StringUtils.replaceOnceIgnoreCase("", "any", null));
+        assertEquals("", StringUtils.replaceOnceIgnoreCase("", "any", "any"));
+
+        assertEquals("FOO", StringUtils.replaceOnceIgnoreCase("FOO", "", "any"));
+        assertEquals("FOO", StringUtils.replaceOnceIgnoreCase("FOO", null, "any"));
+        assertEquals("FOO", StringUtils.replaceOnceIgnoreCase("FOO", "F", null));
+        assertEquals("FOO", StringUtils.replaceOnceIgnoreCase("FOO", null, null));
+
+        assertEquals("foofoo", StringUtils.replaceOnceIgnoreCase("foofoofoo", "foo", ""));
+
+        // Ignore Case
+        assertEquals("Foofoo", StringUtils.replaceOnceIgnoreCase("FoOFoofoo", "foo", ""));
+    }
+
     /**
      * Test method for 'StringUtils.replaceEach(String, String[], String[])'
      */
@@ -2392,6 +2479,42 @@ public class StringUtilsTest {
         assertEquals("queued", StringUtils.remove("queued", "zz"));
     }
 
+    @Test
+    public void testRemoveIgnoreCase_String() {
+        // StringUtils.removeIgnoreCase(null, *) = null
+        assertEquals(null, StringUtils.removeIgnoreCase(null, null));
+        assertEquals(null, StringUtils.removeIgnoreCase(null, ""));
+        assertEquals(null, StringUtils.removeIgnoreCase(null, "a"));
+
+        // StringUtils.removeIgnoreCase("", *) = ""
+        assertEquals("", StringUtils.removeIgnoreCase("", null));
+        assertEquals("", StringUtils.removeIgnoreCase("", ""));
+        assertEquals("", StringUtils.removeIgnoreCase("", "a"));
+
+        // StringUtils.removeIgnoreCase(*, null) = *
+        assertEquals(null, StringUtils.removeIgnoreCase(null, null));
+        assertEquals("", StringUtils.removeIgnoreCase("", null));
+        assertEquals("a", StringUtils.removeIgnoreCase("a", null));
+
+        // StringUtils.removeIgnoreCase(*, "") = *
+        assertEquals(null, StringUtils.removeIgnoreCase(null, ""));
+        assertEquals("", StringUtils.removeIgnoreCase("", ""));
+        assertEquals("a", StringUtils.removeIgnoreCase("a", ""));
+
+        // StringUtils.removeIgnoreCase("queued", "ue") = "qd"
+        assertEquals("qd", StringUtils.removeIgnoreCase("queued", "ue"));
+
+        // StringUtils.removeIgnoreCase("queued", "zz") = "queued"
+        assertEquals("queued", StringUtils.removeIgnoreCase("queued", "zz"));
+
+        // IgnoreCase
+        // StringUtils.removeIgnoreCase("quEUed", "UE") = "qd"
+        assertEquals("qd", StringUtils.removeIgnoreCase("quEUed", "UE"));
+
+        // StringUtils.removeIgnoreCase("queued", "zZ") = "queued"
+        assertEquals("queued", StringUtils.removeIgnoreCase("queued", "zZ"));
+    }
+
     @Test
     public void testRemove_char() {
         // StringUtils.remove(null, *)       = null
